From 3af16222bbc4cb8b0ce9b2b217682d5232bbcd6d Mon Sep 17 00:00:00 2001
From: Rain Zhang-CA <rainzhang@wonton.(none)>
Date: Tue, 13 May 2014 19:51:24 -0700
Subject: [PATCH 4124/5965] PD #90783-8 HEVC decoder more improvement. Sync
 ucode. Move VPS/SPS/PPS/IDR handling to ARM code for noneseamless play.
 Disable local error handling, enable system error handling.

---
 .../amlogic/amports/m8/ucode/h265/vh265_mc.c  | 893 +++++++++---------
 drivers/amlogic/amports/vh265.c               |  89 +-
 2 files changed, 522 insertions(+), 460 deletions(-)

diff --git a/drivers/amlogic/amports/m8/ucode/h265/vh265_mc.c b/drivers/amlogic/amports/m8/ucode/h265/vh265_mc.c
index 02678036375f..cbbe5101f2f8 100755
--- a/drivers/amlogic/amports/m8/ucode/h265/vh265_mc.c
+++ b/drivers/amlogic/amports/m8/ucode/h265/vh265_mc.c
@@ -20,455 +20,458 @@
  */
 
 #include <linux/types.h>
-
+//project_dos revision 2517
 const u32 vh265_mc[] __attribute__((aligned(8))) = {
 0x06810001,0x06800000,0x0d000001,0x07400040,0x0c000900,0x00000000,
-0x06bffe40,0x07c00000,0x06030400,0x00400000,0x0800c0ff,0x0c02de80,
+0x06bffe40,0x07c00000,0x06030400,0x00400000,0x0800c0ff,0x0c02e300,
 0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,
 0x00000000,0x00000000,0x00000000,0x0c7ffd80,0x00000000,0x00000000,
 0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,
-0x00000000,0x00000000,0x0c7a22c0,0x00000000,0x0c79c3c0,0x06412108,
-0x0c780500,0x00000000,0x0cc00000,0x00400000,0x0800c000,0x08002515,
+0x00000000,0x00000000,0x0c7a1f80,0x00000000,0x0c79c040,0x06412108,
+0x0c780580,0x00000000,0x0cc00000,0x00400000,0x0800c000,0x08002515,
 0x08002608,0x06bfbb80,0x06ffbfc0,0x06070500,0x06070600,0x06070700,
-0x06803fc0,0x07c00140,0x07c00780,0x07c01480,0x0640d000,0x07c03380,
-0x0cc00000,0x00000000,0x06bfff88,0x06030408,0x00400000,0x0640c008,
-0x0a6181c8,0x00000000,0x0c02b100,0x00000000,0x0c79b7c0,0x0800c0ff,
-0x0aa0c048,0x0c781ec0,0x00000000,0x06410108,0x09021208,0x0a6f8008,
-0x00000000,0x0640c808,0x06013408,0x08013503,0x06413508,0x090e1208,
-0x0a6f8008,0x00000000,0x08013500,0x06410c08,0x08010e90,0x09326208,
-0x0aa0c808,0x0c79b280,0x0800c0fe,0x07803348,0x09808008,0x07c03348,
-0x06410c08,0x08010e90,0x09384288,0x09223248,0x07c00189,0x09201248,
-0x07c001c9,0x08010e90,0x07800148,0x0a805288,0x07c0010a,0x0aa0ffc8,
-0x0c023c40,0x05407308,0x0780010a,0x07c0014a,0x06410c08,0x08010e88,
-0x09305248,0x07c00209,0x06410c08,0x08010e90,0x09210248,0x07c00249,
-0x06410c08,0x08010e90,0x09210248,0x07c00289,0x06410c08,0x08010e84,
+0x06803fc0,0x07c00140,0x07c00780,0x07c01480,0x0640d000,0x07c033c0,
+0x0640d200,0x07c03340,0x0cc00000,0x00000000,0x06bfff88,0x06030408,
+0x00400000,0x0640c008,0x0a6181c8,0x00000000,0x0c02b500,0x00000000,
+0x0c79b3c0,0x0800c0ff,0x0aa0c048,0x0c781ec0,0x00000000,0x06410108,
+0x09021208,0x0a6f8008,0x00000000,0x0640c808,0x06013408,0x08013503,
+0x06413508,0x090e1208,0x0a6f8008,0x00000000,0x08013500,0x06410c08,
+0x08010e90,0x09326208,0x0aa0c808,0x0c79ae80,0x0800c0fe,0x07803388,
+0x09808008,0x07c03388,0x06410c08,0x08010e90,0x09384288,0x09223248,
+0x07c00189,0x09201248,0x07c001c9,0x08010e90,0x07800148,0x0a805288,
+0x07c0010a,0x0aa0ffc8,0x0c023880,0x05407308,0x0780010a,0x07c0014a,
+0x06410c08,0x08010e88,0x09305248,0x07c00209,0x06410c08,0x08010e90,
+0x09210248,0x07c00249,0x06410c08,0x08010e90,0x09210248,0x07c00289,
+0x06410c08,0x08010e84,0x08010e90,0x08010e90,0x08010e8c,0x06410c08,
+0x08010e88,0x09308248,0x07c002c9,0x0780018b,0x0aa3800b,0x00000000,
+0x06410c0a,0x08010e90,0x0aa2800b,0x093e124a,0x0aa08009,0x0c780000,
+0x093c124a,0x0aa08009,0x0c780000,0x0540228a,0x0c7ffe00,0x044012cb,
+0x06410c08,0x08010e81,0x093e1248,0x0780018b,0x0ae000cb,0x0a60c009,
+0x0680000a,0x0780018a,0x0ac0a2ca,0x00000000,0x0c019ec0,0x0400c24a,
+0x0c019e40,0x0401024a,0x0c019dc0,0x0401424a,0x0c7ffe00,0x0400128a,
+0x08010e86,0x06806008,0x06010e08,0x06410f08,0x0befc0c8,0x09118208,
+0x0aa08008,0x0c780000,0x06410c08,0x08010e81,0x093e1248,0x0aa40009,
+0x08010e90,0x08010e90,0x08010e90,0x08010e90,0x06410c08,0x08010e81,
+0x093e1248,0x0aa0c009,0x0c0197c0,0x06800049,0x0c019740,0x06800049,
+0x0aa0c008,0x0c780000,0x00000000,0x08010e81,0x0c7994c0,0x0800c0ff,
+0x0aa0c088,0x0c78a440,0x00000000,0x06410c08,0x08010e90,0x09326208,
+0x0aa0c848,0x0c799280,0x0800c0fe,0x07803388,0x09808048,0x07c03388,
+0x06410c08,0x08010e88,0x09384288,0x09323248,0x07c00609,0x09301248,
+0x07c00649,0x07800148,0x0a808288,0x07c0010a,0x0c021d00,0x05407308,
+0x0780010a,0x07c0014a,0x0c022880,0x0540730a,0x06410c08,0x08010e88,
+0x09305248,0x07c00689,0x06410c08,0x08010e90,0x09210248,0x07c006c9,
+0x06410c08,0x08010e90,0x09210248,0x07c00709,0x06410c08,0x08010e84,
 0x08010e90,0x08010e90,0x08010e8c,0x06410c08,0x08010e88,0x09308248,
-0x07c002c9,0x0780018b,0x0aa3800b,0x00000000,0x06410c0a,0x08010e90,
+0x07c00749,0x0780060b,0x0aa3800b,0x00000000,0x06410c0a,0x08010e90,
 0x0aa2800b,0x093e124a,0x0aa08009,0x0c780000,0x093c124a,0x0aa08009,
-0x0c780000,0x0540228a,0x0c7ffe00,0x044012cb,0x06410c08,0x08010e81,
-0x093e1248,0x0780018b,0x0ae000cb,0x0a60c009,0x0680000a,0x0780018a,
-0x0ac0a2ca,0x00000000,0x0c01a2c0,0x0400c24a,0x0c01a240,0x0401024a,
-0x0c01a1c0,0x0401424a,0x0c7ffe00,0x0400128a,0x08010e86,0x06806008,
-0x06010e08,0x06410f08,0x0befc0c8,0x09118208,0x0aa08008,0x0c780000,
-0x06410c08,0x08010e81,0x093e1248,0x0aa40009,0x08010e90,0x08010e90,
-0x08010e90,0x08010e90,0x06410c08,0x08010e81,0x093e1248,0x0aa0c009,
-0x0c019bc0,0x06800049,0x0c019b40,0x06800049,0x0aa0c008,0x0c780000,
-0x00000000,0x08010e81,0x0c7998c0,0x0800c0ff,0x0aa0c088,0x0c78a440,
-0x00000000,0x06410c08,0x08010e90,0x09326208,0x0aa0c848,0x0c799680,
-0x0800c0fe,0x07803348,0x09808048,0x07c03348,0x06410c08,0x08010e88,
-0x09384288,0x09323248,0x07c00609,0x09301248,0x07c00649,0x07800148,
-0x0a808288,0x07c0010a,0x0c0220c0,0x05407308,0x0780010a,0x07c0014a,
-0x0c022c40,0x0540730a,0x06410c08,0x08010e88,0x09305248,0x07c00689,
-0x06410c08,0x08010e90,0x09210248,0x07c006c9,0x06410c08,0x08010e90,
-0x09210248,0x07c00709,0x06410c08,0x08010e84,0x08010e90,0x08010e90,
-0x08010e8c,0x06410c08,0x08010e88,0x09308248,0x07c00749,0x0780060b,
-0x0aa3800b,0x00000000,0x06410c0a,0x08010e90,0x0aa2800b,0x093e124a,
-0x0aa08009,0x0c780000,0x093c124a,0x0aa08009,0x0c780000,0x0540228a,
-0x0c7ffe00,0x044012cb,0x0c018940,0x06800049,0x07800789,0x0a805209,
-0x07c00108,0x0aa0ffc9,0x0c021b00,0x05407309,0x07800108,0x07c00788,
-0x0c0186c0,0x068007c9,0x0c018640,0x06800809,0x0c0185c0,0x06800849,
-0x07800808,0x07800849,0x09610209,0x06012308,0x06410c08,0x08010e81,
-0x093e1248,0x0aa24009,0x0c018340,0x06800049,0x0c0182c0,0x06800049,
-0x0c018240,0x06800049,0x0c0181c0,0x06800049,0x0c018140,0x06800889,
-0x0c0180c0,0x068008c9,0x0c018040,0x06800909,0x06410c08,0x08010e81,
-0x093e1248,0x0780060b,0x0ae000cb,0x0a60c009,0x0680000a,0x0780060a,
-0x0ac0a2ca,0x00000000,0x0c017d40,0x0402524a,0x0c017cc0,0x0402924a,
-0x0c017c40,0x0402d24a,0x0c7ffe00,0x0400128a,0x0c017b40,0x06800c49,
-0x07800c49,0x07c00cc9,0x0c017a40,0x06800c89,0x07800c49,0x07800c88,
-0x02009209,0x07c00d09,0x07800d08,0x0a6300c8,0x07800809,0x0403f249,
-0x05806249,0x07c00dc9,0x07800849,0x0403f249,0x05806249,0x07c00e09,
-0x08040640,0x0c780900,0x068210e0,0x0a630088,0x07800809,0x0401f249,
-0x05805249,0x07c00dc9,0x07800849,0x0401f249,0x05805249,0x07c00e09,
-0x08040620,0x0c780600,0x06821660,0x0a630048,0x07800809,0x0400f249,
-0x05804249,0x07c00dc9,0x07800849,0x0400f249,0x05804249,0x07c00e09,
-0x08040610,0x0c780300,0x06821ea0,0x07800809,0x04007249,0x05803249,
-0x07c00dc9,0x07800849,0x04007249,0x05803249,0x07c00e09,0x08040608,
-0x068229e0,0x07800dc8,0x07800e09,0x0e000248,0x00000000,0x00000000,
-0x06800009,0x0f000200,0x04401208,0x0aa10008,0x05801208,0x0c7fff80,
-0x04001249,0x07c01409,0x07800809,0x0780084a,0x0960f24a,0x06040209,
-0x07800dc9,0x07800e0a,0x0960c24a,0x06040309,0x08040400,0x0c016780,
-0x06800049,0x07800049,0x07c00d49,0x0c016680,0x06800089,0x07800049,
-0x07800088,0x02009209,0x07c00d89,0x0c016500,0x06800e89,0x0c016480,
-0x06800ec9,0x06410c08,0x08010e81,0x093e1248,0x0aa38009,0x07c00f09,
-0x06410c08,0x08010e81,0x093e1248,0x0aa0c009,0x0c0016c0,0x00000000,
-0x07800788,0x0aa0c008,0x068f000c,0x069f000c,0x0c020cc0,0x00000000,
-0x06410c08,0x08010e83,0x093e1248,0x07c00f49,0x093c1248,0x07c00f89,
-0x093a1248,0x0aa58009,0x06800009,0x06410c08,0x08010e88,0x09384248,
-0x04001249,0x07c01009,0x09304248,0x04001249,0x07c01049,0x0c015b80,
-0x06801089,0x0c015b00,0x068010c9,0x07801089,0x02008248,0x07c010c8,
-0x06410c08,0x08010e81,0x093e1248,0x07c01109,0x06800049,0x07c00fc9,
-0x0c015840,0x06801149,0x0680000e,0x0780114f,0x044013cf,0x0aa1000f,
-0x058013cf,0x0c7fff80,0x0400138e,0x07c0118e,0x0780114f,0x0680000e,
-0x0b4143ce,0x00000000,0x0480f20e,0x05404208,0x0680800d,0x0200d20d,
-0x0c002580,0x00000000,0x0480f20e,0x0aa1c3c8,0x00000000,0x0400120e,
-0x0b4043c8,0x00000000,0x0c780100,0x00000000,0x0c004600,0x00000000,
-0x0c7ffb80,0x0400138e,0x06410c08,0x08010e81,0x093e1248,0x07c011c9,
-0x0aa10009,0x0c014f00,0x06801209,0x0a600008,0x06410c08,0x08010e83,
-0x093e1248,0x07c01249,0x093c1248,0x07c01289,0x093a1248,0x0aa0c009,
-0x0c004a80,0x00000000,0x08010e81,0x0c794a80,0x0800c0ff,0x068c8408,
-0x06c41508,0x06070508,0x06070608,0x06070708,0x0680000b,0x0680000a,
-0x06410c08,0x08010e81,0x093e1248,0x0aa08009,0x0c780f00,0x0c014800,
-0x06800049,0x0aa140cb,0x0400e3ca,0x0ba0c0ca,0x0680038f,0x068003cf,
-0x0a60c00b,0x0c780180,0x0647050d,0x0a60c04b,0x0c7800c0,0x0647060d,
-0x0647070d,0x0aa38008,0x0400038f,0x0aa300cb,0x0680000e,0x0240f20a,
-0x0aa2410f,0x0920438d,0x0aa1c0cf,0x0918438d,0x0aa1408f,0x0910438d,
-0x0aa0c04f,0x0908438d,0x0900438d,0x0a61c0cb,0x0a60c00a,0x0c7800c0,
-0x0970434e,0x0978434e,0x0c780ec0,0x0607070d,0x0aa1c00a,0x0aa2004a,
-0x0aa2408a,0x0aa280ca,0x0aa2c10a,0x0c7802c0,0x0968434e,0x0c780240,
-0x0940434e,0x0c7801c0,0x0948434e,0x0c780140,0x0950434e,0x0c7800c0,
-0x0958434e,0x0960434e,0x0aa1000b,0x0aa1404b,0x0c7809c0,0x0607070d,
-0x0c780940,0x0607050d,0x0c7808c0,0x0607060d,0x0ba2c08b,0x0680020c,
-0x0c013a80,0x06800049,0x04008308,0x0aa0c08b,0x0680f808,0x0680f988,
-0x02008288,0x06070208,0x0607040c,0x0aa1c00b,0x0680e008,0x0aa1404b,
-0x06800008,0x0aa0c08b,0x06806008,0x0680c008,0x0680040e,0x0aa1000b,
-0x0540424a,0x0680100e,0x0540624a,0x02008248,0x06070208,0x0680004d,
-0x0c013480,0x06800049,0x0200c20c,0x048ff30c,0x0607040c,0x0b8fb38d,
-0x0400134d,0x0ba0c0cb,0x06800149,0x06800049,0x0b40324a,0x0c7fe6c0,
-0x0400128a,0x0b60c0cb,0x0c7fe5c0,0x040012cb,0x0cc00000,0x00000000,
-0x0aa1400e,0x06800009,0x06410c08,0x08010e81,0x093e1248,0x0a608009,
-0x0c781540,0x07801149,0x0a40624e,0x0440144e,0x0c012bc0,0x06800049,
-0x0c002500,0x02411211,0x0480f451,0x05404451,0x06808008,0x02011211,
-0x06800010,0x06800012,0x06410c08,0x08010e81,0x093e1288,0x0c012880,
-0x06800049,0x0aa1000a,0x040014c8,0x06800009,0x024134c9,0x0b60c412,
-0x06a0000b,0x0700044b,0x04001451,0x04001492,0x0200c2d3,0x091e12cb,
-0x0900e30c,0x06410c08,0x08010e81,0x093e1288,0x095c130a,0x0a62000a,
-0x06410c08,0x08010e81,0x093e1288,0x0a61000a,0x0a62400b,0x0c7ffb80,
-0x00000000,0x0740034c,0x0400134d,0x04001410,0x0a60c00b,0x0c7ffa00,
-0x00000000,0x0b60c410,0x06a00008,0x07400348,0x0ae08050,0x0c780780,
-0x0440134d,0x07000351,0x0900e451,0x091a1491,0x06900013,0x0aa0c012,
-0x02012453,0x02412453,0x0680004c,0x0b41240c,0x0241130d,0x07000451,
-0x0900e451,0x091a12d1,0x06900013,0x0aa0c00b,0x0200b453,0x0240b453,
-0x0b4072d2,0x0241130d,0x0400048b,0x0700044b,0x07000353,0x0740034b,
-0x07400453,0x0c7ffc00,0x0400130c,0x0c7ff8c0,0x04401410,0x0cc00000,
-0x00000000,0x0c011780,0x06800049,0x0c011700,0x06800089,0x06800010,
-0x0680000a,0x0680000b,0x0780004c,0x0b41130b,0x0c011540,0x068000c9,
-0x0240a20a,0x0440128a,0x06410c08,0x08010e81,0x093e1248,0x0900e20a,
-0x095c1209,0x07400348,0x0400134d,0x04001410,0x0bacc410,0x040012cb,
-0x0cc00000,0x00000000,0x0680000a,0x0680000b,0x0780008c,0x0b41130b,
-0x0c011040,0x068000c9,0x0200a20a,0x0400128a,0x06410c08,0x08010e81,
+0x0c780000,0x0540228a,0x0c7ffe00,0x044012cb,0x0c018540,0x06800049,
+0x07800789,0x0a805209,0x07c00108,0x0aa0ffc9,0x0c021740,0x05407309,
+0x07800108,0x07c00788,0x0c0182c0,0x068007c9,0x0c018240,0x06800809,
+0x0c0181c0,0x06800849,0x07800808,0x07800849,0x09610209,0x06012308,
+0x06410c08,0x08010e81,0x093e1248,0x0aa24009,0x0c017f40,0x06800049,
+0x0c017ec0,0x06800049,0x0c017e40,0x06800049,0x0c017dc0,0x06800049,
+0x0c017d40,0x06800889,0x0c017cc0,0x068008c9,0x0c017c40,0x06800909,
+0x06410c08,0x08010e81,0x093e1248,0x0780060b,0x0ae000cb,0x0a60c009,
+0x0680000a,0x0780060a,0x0ac0a2ca,0x00000000,0x0c017940,0x0402524a,
+0x0c0178c0,0x0402924a,0x0c017840,0x0402d24a,0x0c7ffe00,0x0400128a,
+0x0c017740,0x06800c49,0x07800c49,0x07c00cc9,0x0c017640,0x06800c89,
+0x07800c49,0x07800c88,0x02009209,0x07c00d09,0x07800d08,0x0a6300c8,
+0x07800809,0x0403f249,0x05806249,0x07c00dc9,0x07800849,0x0403f249,
+0x05806249,0x07c00e09,0x08040640,0x0c780900,0x06820da0,0x0a630088,
+0x07800809,0x0401f249,0x05805249,0x07c00dc9,0x07800849,0x0401f249,
+0x05805249,0x07c00e09,0x08040620,0x0c780600,0x06821320,0x0a630048,
+0x07800809,0x0400f249,0x05804249,0x07c00dc9,0x07800849,0x0400f249,
+0x05804249,0x07c00e09,0x08040610,0x0c780300,0x06821b60,0x07800809,
+0x04007249,0x05803249,0x07c00dc9,0x07800849,0x04007249,0x05803249,
+0x07c00e09,0x08040608,0x068226a0,0x07800dc8,0x07800e09,0x0e000248,
+0x00000000,0x00000000,0x06800009,0x0f000200,0x04401208,0x0aa10008,
+0x05801208,0x0c7fff80,0x04001249,0x07c01409,0x07800809,0x0780084a,
+0x0960f24a,0x06040209,0x07800dc9,0x07800e0a,0x0960c24a,0x06040309,
+0x08040400,0x0c016380,0x06800049,0x07800049,0x07c00d49,0x0c016280,
+0x06800089,0x07800049,0x07800088,0x02009209,0x07c00d89,0x0c016100,
+0x06800e89,0x0c016080,0x06800ec9,0x06410c08,0x08010e81,0x093e1248,
+0x0aa38009,0x07c00f09,0x06410c08,0x08010e81,0x093e1248,0x0aa0c009,
+0x0c0016c0,0x00000000,0x07800788,0x0aa0c008,0x068f000c,0x069f000c,
+0x0c020900,0x00000000,0x06410c08,0x08010e83,0x093e1248,0x07c00f49,
+0x093c1248,0x07c00f89,0x093a1248,0x0aa58009,0x06800009,0x06410c08,
+0x08010e88,0x09384248,0x04001249,0x07c01009,0x09304248,0x04001249,
+0x07c01049,0x0c015780,0x06801089,0x0c015700,0x068010c9,0x07801089,
+0x02008248,0x07c010c8,0x06410c08,0x08010e81,0x093e1248,0x07c01109,
+0x06800049,0x07c00fc9,0x0c015440,0x06801149,0x0680000e,0x0780114f,
+0x044013cf,0x0aa1000f,0x058013cf,0x0c7fff80,0x0400138e,0x07c0118e,
+0x0780114f,0x0680000e,0x0b4143ce,0x00000000,0x0480f20e,0x05404208,
+0x0680800d,0x0200d20d,0x0c002580,0x00000000,0x0480f20e,0x0aa1c3c8,
+0x00000000,0x0400120e,0x0b4043c8,0x00000000,0x0c780100,0x00000000,
+0x0c004600,0x00000000,0x0c7ffb80,0x0400138e,0x06410c08,0x08010e81,
+0x093e1248,0x07c011c9,0x0aa10009,0x0c014b00,0x06801209,0x0a600008,
+0x06410c08,0x08010e83,0x093e1248,0x07c01249,0x093c1248,0x07c01289,
+0x093a1248,0x0aa0c009,0x0c004a80,0x00000000,0x08010e81,0x0c794680,
+0x0800c0ff,0x068c8408,0x06c41508,0x06070508,0x06070608,0x06070708,
+0x0680000b,0x0680000a,0x06410c08,0x08010e81,0x093e1248,0x0aa08009,
+0x0c780f00,0x0c014400,0x06800049,0x0aa140cb,0x0400e3ca,0x0ba0c0ca,
+0x0680038f,0x068003cf,0x0a60c00b,0x0c780180,0x0647050d,0x0a60c04b,
+0x0c7800c0,0x0647060d,0x0647070d,0x0aa38008,0x0400038f,0x0aa300cb,
+0x0680000e,0x0240f20a,0x0aa2410f,0x0920438d,0x0aa1c0cf,0x0918438d,
+0x0aa1408f,0x0910438d,0x0aa0c04f,0x0908438d,0x0900438d,0x0a61c0cb,
+0x0a60c00a,0x0c7800c0,0x0970434e,0x0978434e,0x0c780ec0,0x0607070d,
+0x0aa1c00a,0x0aa2004a,0x0aa2408a,0x0aa280ca,0x0aa2c10a,0x0c7802c0,
+0x0968434e,0x0c780240,0x0940434e,0x0c7801c0,0x0948434e,0x0c780140,
+0x0950434e,0x0c7800c0,0x0958434e,0x0960434e,0x0aa1000b,0x0aa1404b,
+0x0c7809c0,0x0607070d,0x0c780940,0x0607050d,0x0c7808c0,0x0607060d,
+0x0ba2c08b,0x0680020c,0x0c013680,0x06800049,0x04008308,0x0aa0c08b,
+0x0680f808,0x0680f988,0x02008288,0x06070208,0x0607040c,0x0aa1c00b,
+0x0680e008,0x0aa1404b,0x06800008,0x0aa0c08b,0x06806008,0x0680c008,
+0x0680040e,0x0aa1000b,0x0540424a,0x0680100e,0x0540624a,0x02008248,
+0x06070208,0x0680004d,0x0c013080,0x06800049,0x0200c20c,0x048ff30c,
+0x0607040c,0x0b8fb38d,0x0400134d,0x0ba0c0cb,0x06800149,0x06800049,
+0x0b40324a,0x0c7fe6c0,0x0400128a,0x0b60c0cb,0x0c7fe5c0,0x040012cb,
+0x0cc00000,0x00000000,0x0aa1400e,0x06800009,0x06410c08,0x08010e81,
+0x093e1248,0x0a608009,0x0c781540,0x07801149,0x0a40624e,0x0440144e,
+0x0c0127c0,0x06800049,0x0c002500,0x02411211,0x0480f451,0x05404451,
+0x06808008,0x02011211,0x06800010,0x06800012,0x06410c08,0x08010e81,
+0x093e1288,0x0c012480,0x06800049,0x0aa1000a,0x040014c8,0x06800009,
+0x024134c9,0x0b60c412,0x06a0000b,0x0700044b,0x04001451,0x04001492,
+0x0200c2d3,0x091e12cb,0x0900e30c,0x06410c08,0x08010e81,0x093e1288,
+0x095c130a,0x0a62000a,0x06410c08,0x08010e81,0x093e1288,0x0a61000a,
+0x0a62400b,0x0c7ffb80,0x00000000,0x0740034c,0x0400134d,0x04001410,
+0x0a60c00b,0x0c7ffa00,0x00000000,0x0b60c410,0x06a00008,0x07400348,
+0x0ae08050,0x0c780780,0x0440134d,0x07000351,0x0900e451,0x091a1491,
+0x06900013,0x0aa0c012,0x02012453,0x02412453,0x0680004c,0x0b41240c,
+0x0241130d,0x07000451,0x0900e451,0x091a12d1,0x06900013,0x0aa0c00b,
+0x0200b453,0x0240b453,0x0b4072d2,0x0241130d,0x0400048b,0x0700044b,
+0x07000353,0x0740034b,0x07400453,0x0c7ffc00,0x0400130c,0x0c7ff8c0,
+0x04401410,0x0cc00000,0x00000000,0x0c011380,0x06800049,0x0c011300,
+0x06800089,0x06800010,0x0680000a,0x0680000b,0x0780004c,0x0b41130b,
+0x0c011140,0x068000c9,0x0240a20a,0x0440128a,0x06410c08,0x08010e81,
 0x093e1248,0x0900e20a,0x095c1209,0x07400348,0x0400134d,0x04001410,
-0x0bacc410,0x040012cb,0x0cc00000,0x00000000,0x06a00008,0x07400348,
-0x0cc00000,0x00000000,0x0580424e,0x07c01449,0x05409249,0x0640c208,
-0x02008248,0x06035108,0x06808008,0x06035208,0x06a08008,0x06035008,
-0x06435008,0x0580f208,0x0bef8048,0x00000000,0x06435308,0x0900c208,
-0x0a6f8008,0x00000000,0x0cc00000,0x00000000,0x05804251,0x07801448,
-0x0a80e209,0x07c01449,0x05409249,0x0640c208,0x02008248,0x06035108,
-0x06808008,0x06035208,0x06b08008,0x06035008,0x06435008,0x0580f208,
-0x0bef8048,0x00000000,0x0cc00000,0x00000000,0x06410c08,0x08010e81,
-0x093e1248,0x0aa1c009,0x00000000,0x06410c08,0x08010e88,0x09301248,
-0x07c012c9,0x0aa03fc9,0x06410c08,0x08010e81,0x093e1248,0x0aa08009,
-0x0c780000,0x06410c08,0x08010e81,0x093e1248,0x0aa08009,0x0c780000,
-0x06410c08,0x08010e81,0x093e1248,0x0aa08009,0x0c780000,0x06410c08,
-0x08010e81,0x093e1248,0x093c1248,0x093a1248,0x09381248,0x0aa24009,
-0x0c00f9c0,0x06801309,0x0c00f940,0x06801349,0x0c00f8c0,0x06801389,
-0x0c00f840,0x068013c9,0x06410c08,0x08010e81,0x093e1248,0x0aa08009,
-0x0c780000,0x06410c08,0x08010e81,0x093e1248,0x0aa08009,0x0c780000,
-0x0cc00000,0x00000000,0x0aa0c0c8,0x0c782f80,0x00000000,0x06410c08,
-0x08010e90,0x09326208,0x0aa0c888,0x0c78f200,0x0800c0fe,0x07803348,
-0x09808088,0x07c03348,0x0c00f1c0,0x06800049,0x07801489,0x0a805209,
-0x07c00108,0x0aa0ffc9,0x0c019440,0x05407309,0x07800108,0x07c01488,
-0x0c00ef40,0x068014c9,0x0780078a,0x0a808288,0x07c00108,0x0c018140,
-0x0540730a,0x0780010a,0x07c0078a,0x0c018bc0,0x0540730a,0x06410c08,
-0x08010e87,0x093e1248,0x07c01509,0x093c1248,0x07c01549,0x09363248,
-0x07c01589,0x09341248,0x07c015c9,0x09321248,0x07c01609,0x0c00e980,
-0x06801649,0x04001208,0x07c01648,0x0c00e880,0x06801689,0x04001208,
-0x07c01688,0x0c00e940,0x068016c9,0x0401a208,0x07c016c8,0x06410c08,
-0x08010e83,0x093e1248,0x07c01709,0x093c1248,0x07c01749,0x093a1248,
-0x07c01789,0x0aa10009,0x06800008,0x0c00e400,0x068017c9,0x07c017c8,
-0x0c00e500,0x06801809,0x0c00e480,0x06801849,0x06410c08,0x08010e86,
-0x093e1248,0x07c01889,0x093c1248,0x07c018c9,0x093a1248,0x07c01909,
-0x09381248,0x07c01949,0x09361248,0x07c01989,0x09341248,0x07c01cc9,
-0x0aa0c009,0x0c0141c0,0x00000000,0x06800048,0x07c01c88,0x07801988,
-0x0a608008,0x0c780a80,0x0c00dcc0,0x06801a09,0x07800dc8,0x07c01a88,
-0x0c00dbc0,0x06801a49,0x07800e08,0x07c01b88,0x06410c08,0x08010e81,
-0x093e1248,0x07c019c9,0x0aa14009,0x0c014600,0x00000000,0x0c780580,
-0x00000000,0x07801a0b,0x0ae0010b,0x0680000a,0x0b4072ca,0x0c00d780,
-0x0406a24a,0x04001208,0x07400248,0x0c7ffec0,0x0400128a,0x07801a4b,
-0x0ae0010b,0x0680000a,0x0b4072ca,0x0c00d500,0x0406e24a,0x04001208,
-0x07400248,0x0c7ffec0,0x0400128a,0x06410c08,0x08010e81,0x093e1248,
-0x07c01c89,0x06410c08,0x08010e82,0x093e1248,0x07c01d09,0x093c1248,
-0x0a618009,0x07c01d49,0x07c01d89,0x07c01dc9,0x0c780340,0x07c01e09,
-0x06410c08,0x08010e82,0x093e1248,0x07c01d49,0x093c1248,0x0a618009,
-0x07c01d89,0x0c00cfc0,0x06801dc9,0x0c00cf40,0x06801e09,0x06410c08,
-0x08010e81,0x093e1248,0x0a628009,0x07800f08,0x0aa28008,0x078014c8,
-0x0aa0c008,0x068f000c,0x069f000c,0x0c0189c0,0x00000000,0x0c7800c0,
-0x0c077ec0,0x00000000,0x06410c08,0x08010e81,0x093e1248,0x07c01e49,
-0x0c00c840,0x06801e89,0x04002208,0x07c01e88,0x06410c08,0x08010e81,
-0x093e1248,0x07c01ec9,0x08010e81,0x0c019000,0x00000000,0x0c78c480,
-0x0800c0ff,0x0aa0c108,0x0c786b40,0x00000000,0x06410c08,0x08010e90,
-0x09326208,0x0b210548,0x07c01f08,0x0c78c200,0x0800c0fe,0x06410c08,
-0x08010e81,0x093e1248,0x07c01f49,0x07801f08,0x0ba1c408,0x06800009,
-0x0ae14548,0x06800009,0x06410c08,0x08010e81,0x093e1248,0x07c01f89,
-0x0c00bf40,0x06800049,0x07801489,0x0a808248,0x07c00108,0x0c016200,
-0x05407309,0x07800108,0x07c01488,0x0c017440,0x05407308,0x078014c8,
-0x07800789,0x0a808248,0x07c00108,0x0c014ec0,0x05407309,0x07800109,
-0x07c00789,0x0c015940,0x05407309,0x07801f48,0x06800009,0x0aa10008,
-0x07c02009,0x0c780440,0x07c01fc9,0x07801508,0x0aa14008,0x06800009,
-0x06410c08,0x08010e81,0x093e1248,0x07c02009,0x07801409,0x06410c08,
-0x0680080a,0x0240a24a,0x098091c9,0x06010e09,0x03808288,0x07c01fc8,
-0x07802008,0x0aa0c008,0x0c784a40,0x00000000,0x0c01b640,0x00000000,
-0x07801588,0x0aa0c008,0x098081c8,0x06010e08,0x0c00b240,0x06802049,
-0x07801548,0x0aa14008,0x06800049,0x06410c08,0x08010e81,0x093e1248,
-0x07c02089,0x07801f09,0x0aa084c9,0x0a610509,0x06800008,0x0c781600,
-0x07c020c8,0x06410c08,0x07800909,0x04004249,0x0680080a,0x0240a24a,
-0x098091c9,0x06010e09,0x03808288,0x07c020c8,0x06410c08,0x08010e81,
-0x093e1248,0x0a614009,0x0680400d,0x0c077cc0,0x0780114e,0x0c780600,
-0x07801189,0x06410c08,0x068007ca,0x0240a24a,0x098091c9,0x06010e09,
-0x03808288,0x05801208,0x07c02108,0x0c07a1c0,0x04000448,0x0480f211,
-0x05404208,0x06808009,0x02009209,0x0680400d,0x06800008,0x0700024a,
-0x0740034a,0x0400134d,0x04001208,0x0baf0408,0x04001249,0x078011c8,
-0x0a608008,0x0c7808c0,0x0780120a,0x0aa1400a,0x0680000b,0x0c00a300,
-0x06800049,0x040002c8,0x0c00a240,0x06800049,0x0200c20b,0x0aa6400c,
-0x0ac052cc,0x07801208,0x0ba50048,0x0c780000,0x00000000,0x06410c08,
-0x07800909,0x04004249,0x0680080a,0x0240a24a,0x098091c9,0x06010e09,
-0x03808288,0x06410c08,0x08010e81,0x093e1248,0x06410c08,0x08010e81,
-0x093e1248,0x0aa0c009,0x0c009c40,0x06800049,0x0c7ffa40,0x0440130c,
-0x0c0196c0,0x00000000,0x07801248,0x0aa14008,0x06800009,0x06410c08,
-0x08010e81,0x093e1248,0x07c02149,0x07800f88,0x0a610008,0x07c02188,
-0x0c780200,0x07c021c8,0x06410c08,0x08010e82,0x093e1248,0x07c02189,
-0x093c1248,0x07c021c9,0x0780204b,0x0a60c08b,0x0c781700,0x00000000,
-0x0641170a,0x06410c08,0x08010e81,0x093e1248,0x0a628009,0x07801649,
-0x09546289,0x07c02209,0x0a60c00b,0x06800009,0x07801689,0x09606289,
-0x0c7803c0,0x07c02249,0x0c0091c0,0x06802209,0x07802209,0x04001249,
-0x09546289,0x07c02209,0x0aa1c04b,0x0c009000,0x06802249,0x07802249,
-0x04001249,0x09606289,0x07c02249,0x0601170a,0x06800009,0x07c02989,
-0x07801e49,0x0aa14009,0x07802949,0x0b20c049,0x0c0192c0,0x00000000,
-0x0a61400b,0x06800009,0x06410c08,0x08010e81,0x093e1248,0x0641170a,
-0x096c1289,0x0601170a,0x07c02289,0x07801609,0x0aa14009,0x06800009,
-0x06410c08,0x08010e81,0x093e1248,0x07c022c9,0x06411008,0x09581209,
-0x06011008,0x07802149,0x0aa48009,0x0a61400b,0x06800049,0x06410c08,
-0x08010e81,0x093e1248,0x07c02309,0x0aa1808b,0x0aa0c009,0x06802249,
-0x06802209,0x07000249,0x0ae10049,0x06800009,0x0c780100,0x07c02349,
-0x0c008340,0x06802349,0x0aa0c04b,0x078018c9,0x07801909,0x06047b09,
-0x08047c00,0x0aa10009,0x08047d00,0x0c006ac0,0x00000000,0x0c008080,
-0x06802389,0x06800149,0x02409209,0x06411708,0x096e3209,0x06011708,
-0x0c008080,0x068023c9,0x078016c8,0x079023c9,0x02008248,0x07c02408,
-0x07802049,0x09446248,0x078021c8,0x09501248,0x07802188,0x09521248,
-0x07800cc8,0x09542248,0x07800d08,0x09582248,0x07800d48,0x095c2248,
-0x07800d88,0x09602248,0x07801808,0x09647248,0x07801848,0x09727248,
-0x06011109,0x06411009,0x09809389,0x06011009,0x07801888,0x0aa3c008,
-0x0c007900,0x06802449,0x0c007880,0x06802489,0x06411109,0x07801808,
-0x0780244a,0x02008288,0x09647248,0x07801848,0x0780248a,0x02008288,
-0x09727248,0x06011109,0x07801d48,0x0aa14008,0x06800009,0x06410c08,
-0x08010e81,0x093e1248,0x07c024c9,0x0a620009,0x07801d89,0x07c02509,
-0x07801dc9,0x07c02549,0x07801e09,0x0c7802c0,0x07c02589,0x06410c08,
-0x08010e81,0x093e1248,0x0a618009,0x07c02509,0x0c007080,0x06802549,
-0x0c007000,0x06802589,0x07801d08,0x0aa34008,0x06800009,0x07802188,
-0x0a61c008,0x078021c8,0x0a614008,0x07802508,0x0aa0c008,0x0c780140,
-0x06800009,0x06410c08,0x08010e81,0x093e1248,0x07c025c9,0x07801988,
-0x0a610008,0x07801cc8,0x0a608008,0x0c780440,0x0c0068c0,0x06802609,
-0x0780260b,0x0aa3400b,0x0c0067c0,0x06800049,0x04001308,0x0680080a,
-0x0240a30a,0x0980c1cc,0x06410c08,0x06010e0c,0x03808288,0x07c00048,
-0x0aef004b,0x044012cb,0x07801ec8,0x0aa20008,0x0c006440,0x06800049,
-0x0aa14008,0x00000000,0x08010e88,0x0c7fff40,0x04401208,0x0640ce09,
-0x0a274089,0x06420408,0x0a2fc048,0x00000000,0x06020408,0x0be3c049,
-0x06800008,0x09808508,0x09808588,0x06000408,0x06800008,0x06000408,
-0x06450008,0x09808008,0x06050008,0x06440008,0x09808008,0x06040008,
-0x09c08008,0x06040008,0x0a21c049,0x06460108,0x0a216008,0x00000000,
-0x06460408,0x0a2fc048,0x00000000,0x06460408,0x06060408,0x0800c008,
-0x08007401,0x0c013080,0x00000000,0x0c785900,0x00000000,0x0aa0c148,
-0x0c7841c0,0x00000000,0x08010e81,0x06812008,0x06010e08,0x00000000,
-0x08010e00,0x07802008,0x0a60c008,0x06800008,0x07c02648,0x07801f48,
-0x0aa58008,0x06800008,0x07c02888,0x07c028c8,0x07c00e48,0x07c02688,
-0x07c026c8,0x07c02708,0x07c02748,0x07801989,0x0aa0c009,0x07800dc8,
-0x07801a88,0x07c02788,0x07c02808,0x0aa0c009,0x07800e08,0x07801b88,
-0x07c027c8,0x07c02848,0x0c7801c0,0x068001df,0x07801fc8,0x07800e49,
-0x0a803248,0x0aa08009,0x0c780000,0x07802008,0x0a61c008,0x00000000,
-0x068001df,0x07802889,0x078028c8,0x09508248,0x07c02a09,0x07802889,
-0x0a614049,0x07801cc9,0x0aa0c009,0x00000000,0x0981f15f,0x07802889,
-0x078028c8,0x0958c248,0x0978425f,0x06011609,0x0c000500,0x00000000,
-0x07801f49,0x0a60c009,0x07800e48,0x0aa24008,0x07800e48,0x09808788,
-0x098087c8,0x06012408,0x0640d608,0x0aa14008,0x04401208,0x0600d608,
-0x0c784640,0x0800c0ff,0x06a00008,0x06011408,0x0c786a40,0x00000000,
-0x07802789,0x07c029c9,0x078028c9,0x0780274a,0x0a40b289,0x0680000c,
-0x0780288b,0x0780270a,0x0a40728b,0x0680000c,0x0a614009,0x0680008c,
-0x0a60c00a,0x0680008c,0x068000cc,0x07802008,0x09441308,0x06460109,
-0x094a324c,0x06060109,0x06420109,0x0944324c,0x07802048,0x09402248,
-0x07800d08,0x04003208,0x09604248,0x07800cc8,0x04003208,0x09684248,
-0x06020109,0x07800808,0x07800849,0x09610209,0x06020508,0x07802708,
-0x07802749,0x09610209,0x06020708,0x06060708,0x07802808,0x07802849,
-0x09610209,0x06020808,0x06060808,0x07800dc8,0x04401208,0x07800e09,
-0x04401249,0x09610209,0x06020608,0x06800048,0x0541f208,0x06020408,
-0x06450a08,0x09005208,0x0a6f8008,0x06800008,0x07801989,0x0a60c009,
-0x06f00008,0x06d00008,0x06050a08,0x07802708,0x07802749,0x0960a209,
-0x07801989,0x097e1209,0x06050608,0x0780280a,0x07802849,0x0960a289,
-0x0605070a,0x09c087c8,0x0780288a,0x078028c9,0x0960a289,0x0a403288,
-0x00000000,0x098087c8,0x06050808,0x078028c9,0x0780274a,0x0a404289,
-0x07802889,0x0780270a,0x0a802289,0x0c780740,0x0640c908,0x0e000248,
-0x0640ca0a,0x00000000,0x0f000240,0x0200a24a,0x0606110a,0x0606150a,
-0x0606140a,0x04020208,0x078028c9,0x0e000248,0x0640cc0a,0x0640cb0c,
-0x0f000340,0x07802688,0x0e000308,0x0200b34a,0x0402028b,0x00000000,
-0x0f000240,0x0200a24a,0x0606120a,0x0606160a,0x02409309,0x0200b24b,
-0x0606130b,0x0606170b,0x06460408,0x06060408,0x06800048,0x0541f208,
-0x06060408,0x07801988,0x06040508,0x07802808,0x07802849,0x09610209,
-0x06040308,0x07802708,0x07802749,0x09610209,0x06040408,0x07802008,
-0x0a638008,0x06804808,0x06043008,0x06443108,0x0900a248,0x0a6f0009,
-0x091e5248,0x0a6e8009,0x00000000,0x06440008,0x09808008,0x06040008,
-0x09c08008,0x06040008,0x08040701,0x07802008,0x0aa4c008,0x07802888,
-0x0a614008,0x07801cc9,0x0aa0c009,0x0c008780,0x00000000,0x06411009,
-0x091c1249,0x0a6f8009,0x00000000,0x06411009,0x098093c9,0x06011009,
-0x06411009,0x091e1249,0x0a6f8009,0x0cc00000,0x00000000,0x06411009,
-0x091c1249,0x0a6f8009,0x00000000,0x07800dc9,0x0ae0c049,0x0c008000,
-0x00000000,0x06411009,0x098093c9,0x06011009,0x08011c00,0x08011e00,
-0x06411009,0x091e1249,0x0a6f8009,0x0cc00000,0x00000000,0x0a60fc08,
-0x00000000,0x08010e90,0x0c7815c0,0x0800c0ff,0x078007cb,0x0c001600,
-0x06800049,0x06047e08,0x0aa1000b,0x0c0016c0,0x06800049,0x06047f08,
-0x08048000,0x0c0002c0,0x0780220a,0x06047c0c,0x0780204a,0x0a61400a,
-0x08048020,0x0c000140,0x0780224a,0x06047d0c,0x0cc00000,0x00000000,
-0x0400034a,0x06410c0e,0x0980d1cd,0x06010e0d,0x0680000c,0x0400034a,
-0x0aa1c00d,0x093e13ce,0x0580130c,0x097e130f,0x0540138e,0x0c7ffec0,
-0x0440134d,0x0680040d,0x0240d28d,0x0380c34c,0x0aa4000b,0x0400034a,
-0x06410c0e,0x0980d1cd,0x06010e0d,0x0400034a,0x0aa1c00d,0x093e13ce,
+0x0bacc410,0x040012cb,0x0cc00000,0x00000000,0x0680000a,0x0680000b,
+0x0780008c,0x0b41130b,0x0c010c40,0x068000c9,0x0200a20a,0x0400128a,
+0x06410c08,0x08010e81,0x093e1248,0x0900e20a,0x095c1209,0x07400348,
+0x0400134d,0x04001410,0x0bacc410,0x040012cb,0x0cc00000,0x00000000,
+0x06a00008,0x07400348,0x0cc00000,0x00000000,0x0580424e,0x07c01449,
+0x05409249,0x0640c208,0x02008248,0x06035108,0x06808008,0x06035208,
+0x06a08008,0x06035008,0x06435008,0x0580f208,0x0bef8048,0x00000000,
+0x06435308,0x0900c208,0x0a6f8008,0x00000000,0x0cc00000,0x00000000,
+0x05804251,0x07801448,0x0a80e209,0x07c01449,0x05409249,0x0640c208,
+0x02008248,0x06035108,0x06808008,0x06035208,0x06b08008,0x06035008,
+0x06435008,0x0580f208,0x0bef8048,0x00000000,0x0cc00000,0x00000000,
+0x06410c08,0x08010e81,0x093e1248,0x0aa1c009,0x00000000,0x06410c08,
+0x08010e88,0x09301248,0x07c012c9,0x0aa03fc9,0x06410c08,0x08010e81,
+0x093e1248,0x0aa08009,0x0c780000,0x06410c08,0x08010e81,0x093e1248,
+0x0aa08009,0x0c780000,0x06410c08,0x08010e81,0x093e1248,0x0aa08009,
+0x0c780000,0x06410c08,0x08010e81,0x093e1248,0x093c1248,0x093a1248,
+0x09381248,0x0aa24009,0x0c00f5c0,0x06801309,0x0c00f540,0x06801349,
+0x0c00f4c0,0x06801389,0x0c00f440,0x068013c9,0x06410c08,0x08010e81,
+0x093e1248,0x0aa08009,0x0c780000,0x06410c08,0x08010e81,0x093e1248,
+0x0aa08009,0x0c780000,0x0cc00000,0x00000000,0x0aa0c0c8,0x0c782f80,
+0x00000000,0x06410c08,0x08010e90,0x09326208,0x0aa0c888,0x0c78ee00,
+0x0800c0fe,0x07803388,0x09808088,0x07c03388,0x0c00edc0,0x06800049,
+0x07801489,0x0a805209,0x07c00108,0x0aa0ffc9,0x0c019080,0x05407309,
+0x07800108,0x07c01488,0x0c00eb40,0x068014c9,0x0780078a,0x0a808288,
+0x07c00108,0x0c017d80,0x0540730a,0x0780010a,0x07c0078a,0x0c018800,
+0x0540730a,0x06410c08,0x08010e87,0x093e1248,0x07c01509,0x093c1248,
+0x07c01549,0x09363248,0x07c01589,0x09341248,0x07c015c9,0x09321248,
+0x07c01609,0x0c00e580,0x06801649,0x04001208,0x07c01648,0x0c00e480,
+0x06801689,0x04001208,0x07c01688,0x0c00e540,0x068016c9,0x0401a208,
+0x07c016c8,0x06410c08,0x08010e83,0x093e1248,0x07c01709,0x093c1248,
+0x07c01749,0x093a1248,0x07c01789,0x0aa10009,0x06800008,0x0c00e000,
+0x068017c9,0x07c017c8,0x0c00e100,0x06801809,0x0c00e080,0x06801849,
+0x06410c08,0x08010e86,0x093e1248,0x07c01889,0x093c1248,0x07c018c9,
+0x093a1248,0x07c01909,0x09381248,0x07c01949,0x09361248,0x07c01989,
+0x09341248,0x07c01cc9,0x0aa0c009,0x0c013e00,0x00000000,0x06800048,
+0x07c01c88,0x07801988,0x0a608008,0x0c780a80,0x0c00d8c0,0x06801a09,
+0x07800dc8,0x07c01a88,0x0c00d7c0,0x06801a49,0x07800e08,0x07c01b88,
+0x06410c08,0x08010e81,0x093e1248,0x07c019c9,0x0aa14009,0x0c014240,
+0x00000000,0x0c780580,0x00000000,0x07801a0b,0x0ae0010b,0x0680000a,
+0x0b4072ca,0x0c00d380,0x0406a24a,0x04001208,0x07400248,0x0c7ffec0,
+0x0400128a,0x07801a4b,0x0ae0010b,0x0680000a,0x0b4072ca,0x0c00d100,
+0x0406e24a,0x04001208,0x07400248,0x0c7ffec0,0x0400128a,0x06410c08,
+0x08010e81,0x093e1248,0x07c01c89,0x06410c08,0x08010e82,0x093e1248,
+0x07c01d09,0x093c1248,0x0a618009,0x07c01d49,0x07c01d89,0x07c01dc9,
+0x0c780340,0x07c01e09,0x06410c08,0x08010e82,0x093e1248,0x07c01d49,
+0x093c1248,0x0a618009,0x07c01d89,0x0c00cbc0,0x06801dc9,0x0c00cb40,
+0x06801e09,0x06410c08,0x08010e81,0x093e1248,0x0a628009,0x07800f08,
+0x0aa28008,0x078014c8,0x0aa0c008,0x068f000c,0x069f000c,0x0c018600,
+0x00000000,0x0c7800c0,0x0c077ec0,0x00000000,0x06410c08,0x08010e81,
+0x093e1248,0x07c01e49,0x0c00c440,0x06801e89,0x04002208,0x07c01e88,
+0x06410c08,0x08010e81,0x093e1248,0x07c01ec9,0x08010e81,0x0c018c40,
+0x00000000,0x0c78c080,0x0800c0ff,0x0aa0c108,0x0c786740,0x00000000,
+0x06410c08,0x08010e90,0x09326208,0x0b210548,0x07c01f08,0x0c78be00,
+0x0800c0fe,0x06410c08,0x08010e81,0x093e1248,0x07c01f49,0x07801f08,
+0x0ba1c408,0x06800009,0x0ae14548,0x06800009,0x06410c08,0x08010e81,
+0x093e1248,0x07c01f89,0x0c00bb40,0x06800049,0x07801489,0x0a808248,
+0x07c00108,0x0c015e40,0x05407309,0x07800108,0x07c01488,0x0c017080,
+0x05407308,0x078014c8,0x07800789,0x0a808248,0x07c00108,0x0c014b00,
+0x05407309,0x07800109,0x07c00789,0x0c015580,0x05407309,0x07801f48,
+0x06800009,0x0aa10008,0x07c02009,0x0c780440,0x07c01fc9,0x07801508,
+0x0aa14008,0x06800009,0x06410c08,0x08010e81,0x093e1248,0x07c02009,
+0x07801409,0x06410c08,0x0680080a,0x0240a24a,0x098091c9,0x06010e09,
+0x03808288,0x07c01fc8,0x07802008,0x0aa0c008,0x0c784a40,0x00000000,
+0x0c01ba40,0x00000000,0x07801588,0x0aa0c008,0x098081c8,0x06010e08,
+0x0c00ae40,0x06802049,0x07801548,0x0aa14008,0x06800049,0x06410c08,
+0x08010e81,0x093e1248,0x07c02089,0x07801f09,0x0aa084c9,0x0a610509,
+0x06800008,0x0c781600,0x07c020c8,0x06410c08,0x07800909,0x04004249,
+0x0680080a,0x0240a24a,0x098091c9,0x06010e09,0x03808288,0x07c020c8,
+0x06410c08,0x08010e81,0x093e1248,0x0a614009,0x0680400d,0x0c077cc0,
+0x0780114e,0x0c780600,0x07801189,0x06410c08,0x068007ca,0x0240a24a,
+0x098091c9,0x06010e09,0x03808288,0x05801208,0x07c02108,0x0c07a1c0,
+0x04000448,0x0480f211,0x05404208,0x06808009,0x02009209,0x0680400d,
+0x06800008,0x0700024a,0x0740034a,0x0400134d,0x04001208,0x0baf0408,
+0x04001249,0x078011c8,0x0a608008,0x0c7808c0,0x0780120a,0x0aa1400a,
+0x0680000b,0x0c009f00,0x06800049,0x040002c8,0x0c009e40,0x06800049,
+0x0200c20b,0x0aa6400c,0x0ac052cc,0x07801208,0x0ba50048,0x0c780000,
+0x00000000,0x06410c08,0x07800909,0x04004249,0x0680080a,0x0240a24a,
+0x098091c9,0x06010e09,0x03808288,0x06410c08,0x08010e81,0x093e1248,
+0x06410c08,0x08010e81,0x093e1248,0x0aa0c009,0x0c009840,0x06800049,
+0x0c7ffa40,0x0440130c,0x0c019ac0,0x00000000,0x07801248,0x0aa14008,
+0x06800009,0x06410c08,0x08010e81,0x093e1248,0x07c02149,0x07800f88,
+0x0a610008,0x07c02188,0x0c780200,0x07c021c8,0x06410c08,0x08010e82,
+0x093e1248,0x07c02189,0x093c1248,0x07c021c9,0x0780204b,0x0a60c08b,
+0x0c781700,0x00000000,0x0641170a,0x06410c08,0x08010e81,0x093e1248,
+0x0a628009,0x07801649,0x09546289,0x07c02209,0x0a60c00b,0x06800009,
+0x07801689,0x09606289,0x0c7803c0,0x07c02249,0x0c008dc0,0x06802209,
+0x07802209,0x04001249,0x09546289,0x07c02209,0x0aa1c04b,0x0c008c00,
+0x06802249,0x07802249,0x04001249,0x09606289,0x07c02249,0x0601170a,
+0x06800009,0x07c02989,0x07801e49,0x0aa14009,0x07802949,0x0b20c049,
+0x0c0196c0,0x00000000,0x0a61400b,0x06800009,0x06410c08,0x08010e81,
+0x093e1248,0x0641170a,0x096c1289,0x0601170a,0x07c02289,0x07801609,
+0x0aa14009,0x06800009,0x06410c08,0x08010e81,0x093e1248,0x07c022c9,
+0x06411008,0x09581209,0x06011008,0x07802149,0x0aa48009,0x0a61400b,
+0x06800049,0x06410c08,0x08010e81,0x093e1248,0x07c02309,0x0aa1808b,
+0x0aa0c009,0x06802249,0x06802209,0x07000249,0x0ae10049,0x06800009,
+0x0c780100,0x07c02349,0x0c007f40,0x06802349,0x0aa0c04b,0x078018c9,
+0x07801909,0x06047b09,0x08047c00,0x0aa10009,0x08047d00,0x0c0066c0,
+0x00000000,0x0c007c80,0x06802389,0x06800149,0x02409209,0x06411708,
+0x096e3209,0x06011708,0x0c007c80,0x068023c9,0x078016c8,0x079023c9,
+0x02008248,0x07c02408,0x07802049,0x09446248,0x078021c8,0x09501248,
+0x07802188,0x09521248,0x07800cc8,0x09542248,0x07800d08,0x09582248,
+0x07800d48,0x095c2248,0x07800d88,0x09602248,0x07801808,0x09647248,
+0x07801848,0x09727248,0x06011109,0x06411009,0x09809389,0x06011009,
+0x07801888,0x0aa3c008,0x0c007500,0x06802449,0x0c007480,0x06802489,
+0x06411109,0x07801808,0x0780244a,0x02008288,0x09647248,0x07801848,
+0x0780248a,0x02008288,0x09727248,0x06011109,0x07801d48,0x0aa14008,
+0x06800009,0x06410c08,0x08010e81,0x093e1248,0x07c024c9,0x0a620009,
+0x07801d89,0x07c02509,0x07801dc9,0x07c02549,0x07801e09,0x0c7802c0,
+0x07c02589,0x06410c08,0x08010e81,0x093e1248,0x0a618009,0x07c02509,
+0x0c006c80,0x06802549,0x0c006c00,0x06802589,0x07801d08,0x0aa34008,
+0x06800009,0x07802188,0x0a61c008,0x078021c8,0x0a614008,0x07802508,
+0x0aa0c008,0x0c780140,0x06800009,0x06410c08,0x08010e81,0x093e1248,
+0x07c025c9,0x07801988,0x0a610008,0x07801cc8,0x0a608008,0x0c780440,
+0x0c0064c0,0x06802609,0x0780260b,0x0aa3400b,0x0c0063c0,0x06800049,
+0x04001308,0x0680080a,0x0240a30a,0x0980c1cc,0x06410c08,0x06010e0c,
+0x03808288,0x07c00048,0x0aef004b,0x044012cb,0x07801ec8,0x0aa20008,
+0x0c006040,0x06800049,0x0aa14008,0x00000000,0x08010e88,0x0c7fff40,
+0x04401208,0x0640ce09,0x0a234089,0x06420408,0x0a2fc048,0x00000000,
+0x06020408,0x06460108,0x0a216008,0x00000000,0x06460408,0x0a2fc048,
+0x00000000,0x06460408,0x06060408,0x0800c008,0x08007401,0x0c013880,
+0x00000000,0x0c785900,0x00000000,0x0aa0c148,0x0c7841c0,0x00000000,
+0x08010e81,0x06812008,0x06010e08,0x00000000,0x08010e00,0x07802008,
+0x0a60c008,0x06800008,0x07c02648,0x07801f48,0x0aa58008,0x06800008,
+0x07c02888,0x07c028c8,0x07c00e48,0x07c02688,0x07c026c8,0x07c02708,
+0x07c02748,0x07801989,0x0aa0c009,0x07800dc8,0x07801a88,0x07c02788,
+0x07c02808,0x0aa0c009,0x07800e08,0x07801b88,0x07c027c8,0x07c02848,
+0x0c7801c0,0x068001df,0x07801fc8,0x07800e49,0x0a803248,0x0aa08009,
+0x0c780000,0x07802008,0x0a61c008,0x00000000,0x068001df,0x07802889,
+0x078028c8,0x09508248,0x07c02a09,0x07802889,0x0a614049,0x07801cc9,
+0x0aa0c009,0x00000000,0x0981f15f,0x07802889,0x078028c8,0x0958c248,
+0x0978425f,0x06011609,0x0c000500,0x00000000,0x07801f49,0x0a60c009,
+0x07800e48,0x0aa24008,0x07800e48,0x09808788,0x098087c8,0x06012408,
+0x0640d608,0x0aa14008,0x04401208,0x0600d608,0x0c784640,0x0800c0ff,
+0x06a00008,0x06011408,0x0c786a80,0x00000000,0x07802789,0x07c029c9,
+0x078028c9,0x0780274a,0x0a40b289,0x0680000c,0x0780288b,0x0780270a,
+0x0a40728b,0x0680000c,0x0a614009,0x0680008c,0x0a60c00a,0x0680008c,
+0x068000cc,0x07802008,0x09441308,0x06460109,0x094a324c,0x06060109,
+0x06420109,0x0944324c,0x07802048,0x09402248,0x07800d08,0x04003208,
+0x09604248,0x07800cc8,0x04003208,0x09684248,0x06020109,0x07800808,
+0x07800849,0x09610209,0x06020508,0x07802708,0x07802749,0x09610209,
+0x06020708,0x06060708,0x07802808,0x07802849,0x09610209,0x06020808,
+0x06060808,0x07800dc8,0x04401208,0x07800e09,0x04401249,0x09610209,
+0x06020608,0x06800048,0x0541f208,0x06020408,0x06450a08,0x09005208,
+0x0a6f8008,0x06800008,0x07801989,0x0a60c009,0x06f00008,0x06d00008,
+0x06050a08,0x07802708,0x07802749,0x0960a209,0x07801989,0x097e1209,
+0x06050608,0x0780280a,0x07802849,0x0960a289,0x0605070a,0x09c087c8,
+0x0780288a,0x078028c9,0x0960a289,0x0a403288,0x00000000,0x098087c8,
+0x06050808,0x078028c9,0x0780274a,0x0a404289,0x07802889,0x0780270a,
+0x0a802289,0x0c780740,0x0640c908,0x0e000248,0x0640ca0a,0x00000000,
+0x0f000240,0x0200a24a,0x0606110a,0x0606150a,0x0606140a,0x04020208,
+0x078028c9,0x0e000248,0x0640cc0a,0x0640cb0c,0x0f000340,0x07802688,
+0x0e000308,0x0200b34a,0x0402028b,0x00000000,0x0f000240,0x0200a24a,
+0x0606120a,0x0606160a,0x02409309,0x0200b24b,0x0606130b,0x0606170b,
+0x06460408,0x06060408,0x06800048,0x0541f208,0x06060408,0x07801988,
+0x06040508,0x07802808,0x07802849,0x09610209,0x06040308,0x07802708,
+0x07802749,0x09610209,0x06040408,0x07802008,0x0a638008,0x06804808,
+0x06043008,0x06443108,0x0900a248,0x0a6f0009,0x091e5248,0x0a6e8009,
+0x00000000,0x06440008,0x09808008,0x06040008,0x09c08008,0x06040008,
+0x08040701,0x07802008,0x0aa4c008,0x07802888,0x0a614008,0x07801cc9,
+0x0aa0c009,0x0c0087c0,0x00000000,0x06411009,0x091c1249,0x0a6f8009,
+0x00000000,0x06411009,0x098093c9,0x06011009,0x06411009,0x091e1249,
+0x0a6f8009,0x0cc00000,0x00000000,0x06411009,0x091c1249,0x0a6f8009,
+0x00000000,0x07800dc9,0x0ae0c049,0x0c008040,0x00000000,0x06411009,
+0x098093c9,0x06011009,0x08011c00,0x08011e00,0x06411009,0x091e1249,
+0x0a6f8009,0x0cc00000,0x00000000,0x0a60fc08,0x00000000,0x08010e90,
+0x0c7815c0,0x0800c0ff,0x078007cb,0x0c001600,0x06800049,0x06047e08,
+0x0aa1000b,0x0c0016c0,0x06800049,0x06047f08,0x08048000,0x0c0002c0,
+0x0780220a,0x06047c0c,0x0780204a,0x0a61400a,0x08048020,0x0c000140,
+0x0780224a,0x06047d0c,0x0cc00000,0x00000000,0x0400034a,0x06410c0e,
+0x0980d1cd,0x06010e0d,0x0680000c,0x0400034a,0x0aa1c00d,0x093e13ce,
 0x0580130c,0x097e130f,0x0540138e,0x0c7ffec0,0x0440134d,0x0680040d,
-0x0240d28d,0x0380c34c,0x0400034c,0x0ae0c00a,0x0cc00000,0x00000000,
-0x0680000e,0x0680000f,0x0900140d,0x0aa1c010,0x0c000900,0x06800049,
-0x09408388,0x0c000840,0x06800049,0x094083c8,0x0920140d,0x0aa34010,
-0x0c000700,0x06800049,0x09508388,0x0c000640,0x06800049,0x0950a3c8,
-0x0c000580,0x06800049,0x09608388,0x0c0004c0,0x06800049,0x0964a3c8,
-0x0604810e,0x0604810f,0x0580134d,0x0c7ff880,0x0440128a,0x00800000,
-0x07800008,0x0c7e4140,0x06030408,0x06806008,0x06010e08,0x06410f08,
-0x0befc0c8,0x09118208,0x0cc00000,0x07400248,0x0680e008,0x06010e08,
-0x06410f08,0x0befc0c8,0x09118208,0x0cc00000,0x07400248,0x06bfff89,
-0x06030409,0x00400000,0x0be0c108,0x0c7ffa80,0x00000000,0x08012104,
-0x07800dc9,0x0ae0c049,0x06411c22,0x04000862,0x0a26891f,0x06011c21,
-0x0a21481f,0x0c005e80,0x00000000,0x0a25411f,0x09c1f15f,0x07801cc9,
-0x0aa48009,0x08011e00,0x0be42022,0x06a907c9,0x06012b09,0x00000000,
-0x00000000,0x06412b09,0x0c005dc0,0x00000000,0x06411009,0x098093c9,
-0x06011009,0x06411009,0x091e1249,0x0a6f8009,0x0c780800,0x00000000,
-0x0a27a022,0x06a907c9,0x06012b09,0x00000000,0x00000000,0x06412b09,
-0x0c010200,0x00000000,0x06411609,0x093e1249,0x0aa48009,0x06800008,
-0x07c02648,0x06411109,0x07802408,0x09446248,0x06011109,0x06411009,
-0x09809389,0x06011009,0x06411e08,0x0be1e008,0x068001df,0x0c07a7c0,
-0x0978479f,0x06420409,0x0c780100,0x06020409,0x0c7feb40,0x0800c0ff,
-0x0601161e,0x08013e00,0x06413f08,0x06413f08,0x06413f08,0x06413f08,
-0x06413f08,0x06413f08,0x06413f08,0x06413f08,0x06413f08,0x06413f08,
-0x06413e08,0x0640d508,0x0a608008,0x0c780a80,0x04401208,0x0600d508,
-0x06410108,0x09021208,0x0a6f8008,0x00000000,0x0640c708,0x06013408,
-0x08013503,0x06413508,0x090e1208,0x0a6f8008,0x00000000,0x08013500,
-0x0640c808,0x06013408,0x08013501,0x06413508,0x090e1208,0x0a6f8008,
-0x00000000,0x08013500,0x06800408,0x06410c09,0x08010e90,0x09210249,
-0x07c0ffc9,0x0aef0048,0x04401208,0x06410108,0x09021208,0x0a6f8008,
-0x00000000,0x0640c708,0x06013408,0x08013501,0x06413508,0x090e1208,
-0x0a6f8008,0x00000000,0x08013500,0x07800e48,0x09808788,0x098087c8,
-0x06012408,0x06a00008,0x06011408,0x07801988,0x0aa0c008,0x0c00ed40,
-0x00000000,0x09c1f09f,0x07800e48,0x04001208,0x07c00e48,0x07802648,
-0x04001208,0x07c02648,0x0a21409f,0x07802809,0x0b803248,0x00000000,
-0x09c1f05f,0x07802888,0x0a618048,0x04001208,0x07801cc9,0x0aa0c009,
-0x00000000,0x0981f15f,0x07802789,0x0b815248,0x07c02888,0x07802708,
-0x07c02888,0x0981f09f,0x07800e48,0x07800dc9,0x02008248,0x07802809,
-0x02408248,0x07c00e48,0x078028c8,0x04001208,0x078027c9,0x0b807248,
-0x07c028c8,0x07801988,0x0aa0c008,0x0c7806c0,0x00000000,0x0981f0df,
-0x078028c8,0x04001208,0x078027c9,0x0b80f248,0x07802888,0x04001208,
-0x07802789,0x0b80b248,0x07800dc9,0x0b807248,0x078028c8,0x04001208,
-0x07800e09,0x0b803248,0x0c7800c0,0x09c1f11f,0x0981f11f,0x0981f0df,
-0x0780289e,0x078028c8,0x0958c788,0x0978479f,0x0cb80020,0x00000000,
-0x06800048,0x07c02008,0x07802688,0x07801a0c,0x0b41a308,0x06801a89,
-0x02009209,0x0700024a,0x0780270b,0x0200b28b,0x07c0270b,0x07c0288b,
-0x0780274b,0x07c028cb,0x04001208,0x0b808308,0x07c02688,0x07800dc9,
-0x07c02789,0x0780270a,0x02409289,0x0c780ac0,0x07c02809,0x04001249,
-0x0700024a,0x07c0280a,0x0780278b,0x0200b28b,0x0c780900,0x07c0278b,
-0x078026c8,0x07801a4c,0x0b802308,0x0c7ff600,0x06800008,0x07c02688,
-0x07c02888,0x07c02708,0x07801a89,0x07c02809,0x07c02789,0x078026c8,
-0x06801b89,0x02009209,0x0700024a,0x0780274b,0x0200b28b,0x07c0274b,
-0x07c028cb,0x04001208,0x0b808308,0x07c026c8,0x07800e09,0x07c027c9,
-0x0780274a,0x02409289,0x0c780200,0x07c02849,0x04001249,0x0700024a,
-0x07c0284a,0x078027cb,0x0200b28b,0x07c027cb,0x07802748,0x07800dc9,
-0x0e000248,0x00000000,0x00000000,0x07802709,0x0f000200,0x02008248,
-0x07c00e48,0x0c7fe780,0x04c177df,0x07802889,0x0780270a,0x0400228a,
+0x0240d28d,0x0380c34c,0x0aa4000b,0x0400034a,0x06410c0e,0x0980d1cd,
+0x06010e0d,0x0400034a,0x0aa1c00d,0x093e13ce,0x0580130c,0x097e130f,
+0x0540138e,0x0c7ffec0,0x0440134d,0x0680040d,0x0240d28d,0x0380c34c,
+0x0400034c,0x0ae0c00a,0x0cc00000,0x00000000,0x0680000e,0x0680000f,
+0x0900140d,0x0aa1c010,0x0c000900,0x06800049,0x09408388,0x0c000840,
+0x06800049,0x094083c8,0x0920140d,0x0aa34010,0x0c000700,0x06800049,
+0x09508388,0x0c000640,0x06800049,0x0950a3c8,0x0c000580,0x06800049,
+0x09608388,0x0c0004c0,0x06800049,0x0964a3c8,0x0604810e,0x0604810f,
+0x0580134d,0x0c7ff880,0x0440128a,0x00800000,0x07800008,0x0c7e44c0,
+0x06030408,0x06806008,0x06010e08,0x06410f08,0x0befc0c8,0x09118208,
+0x0cc00000,0x07400248,0x0680e008,0x06010e08,0x06410f08,0x0befc0c8,
+0x09118208,0x0cc00000,0x07400248,0x06410509,0x06bfff89,0x06030409,
+0x00400000,0x0be0c108,0x0c7ffa40,0x00000000,0x08012104,0x07800dc9,
+0x0ae0c049,0x06411c22,0x04000862,0x0a26891f,0x06011c21,0x0a21481f,
+0x0c005e80,0x00000000,0x0a25411f,0x09c1f15f,0x07801cc9,0x0aa48009,
+0x08011e00,0x0be42022,0x06a907c9,0x06012b09,0x00000000,0x00000000,
+0x06412b09,0x0c005dc0,0x00000000,0x06411009,0x098093c9,0x06011009,
+0x06411009,0x091e1249,0x0a6f8009,0x0c780800,0x00000000,0x0a27a022,
+0x06a907c9,0x06012b09,0x00000000,0x00000000,0x06412b09,0x0c0109c0,
+0x00000000,0x06411609,0x093e1249,0x0aa48009,0x06800008,0x07c02648,
+0x06411109,0x07802408,0x09446248,0x06011109,0x06411009,0x09809389,
+0x06011009,0x06411e08,0x0be1e008,0x068001df,0x0c07a780,0x0978479f,
+0x06420409,0x0c780100,0x06020409,0x0c7feb00,0x0800c0ff,0x0601161e,
+0x08013e00,0x06413f08,0x06413f08,0x06413f08,0x06413f08,0x06413f08,
+0x06413f08,0x06413f08,0x06413f08,0x06413f08,0x06413f08,0x06413e08,
+0x0640d508,0x0a608008,0x0c780a80,0x04401208,0x0600d508,0x06410108,
+0x09021208,0x0a6f8008,0x00000000,0x0640c708,0x06013408,0x08013503,
+0x06413508,0x090e1208,0x0a6f8008,0x00000000,0x08013500,0x0640c808,
+0x06013408,0x08013501,0x06413508,0x090e1208,0x0a6f8008,0x00000000,
+0x08013500,0x06800408,0x06410c09,0x08010e90,0x09210249,0x07c0ffc9,
+0x0aef0048,0x04401208,0x06410108,0x09021208,0x0a6f8008,0x00000000,
+0x0640c708,0x06013408,0x08013501,0x06413508,0x090e1208,0x0a6f8008,
+0x00000000,0x08013500,0x07800e48,0x09808788,0x098087c8,0x06012408,
+0x06a00008,0x06011408,0x07801988,0x0aa0c008,0x0c00f500,0x00000000,
+0x09c1f09f,0x07800e48,0x04001208,0x07c00e48,0x07802648,0x04001208,
+0x07c02648,0x0a21409f,0x07802809,0x0b803248,0x00000000,0x09c1f05f,
+0x07802888,0x0a618048,0x04001208,0x07801cc9,0x0aa0c009,0x00000000,
+0x0981f15f,0x07802789,0x0b815248,0x07c02888,0x07802708,0x07c02888,
+0x0981f09f,0x07800e48,0x07800dc9,0x02008248,0x07802809,0x02408248,
+0x07c00e48,0x078028c8,0x04001208,0x078027c9,0x0b807248,0x07c028c8,
+0x07801988,0x0aa0c008,0x0c7806c0,0x00000000,0x0981f0df,0x078028c8,
+0x04001208,0x078027c9,0x0b80f248,0x07802888,0x04001208,0x07802789,
+0x0b80b248,0x07800dc9,0x0b807248,0x078028c8,0x04001208,0x07800e09,
+0x0b803248,0x0c7800c0,0x09c1f11f,0x0981f11f,0x0981f0df,0x0780289e,
+0x078028c8,0x0958c788,0x0978479f,0x0cb80020,0x00000000,0x06800048,
+0x07c02008,0x07802688,0x07801a0c,0x0b41a308,0x06801a89,0x02009209,
+0x0700024a,0x0780270b,0x0200b28b,0x07c0270b,0x07c0288b,0x0780274b,
+0x07c028cb,0x04001208,0x0b808308,0x07c02688,0x07800dc9,0x07c02789,
+0x0780270a,0x02409289,0x0c780ac0,0x07c02809,0x04001249,0x0700024a,
+0x07c0280a,0x0780278b,0x0200b28b,0x0c780900,0x07c0278b,0x078026c8,
+0x07801a4c,0x0b802308,0x0c7ff600,0x06800008,0x07c02688,0x07c02888,
+0x07c02708,0x07801a89,0x07c02809,0x07c02789,0x078026c8,0x06801b89,
+0x02009209,0x0700024a,0x0780274b,0x0200b28b,0x07c0274b,0x07c028cb,
+0x04001208,0x0b808308,0x07c026c8,0x07800e09,0x07c027c9,0x0780274a,
+0x02409289,0x0c780200,0x07c02849,0x04001249,0x0700024a,0x07c0284a,
+0x078027cb,0x0200b28b,0x07c027cb,0x07802748,0x07800dc9,0x0e000248,
+0x00000000,0x00000000,0x07802709,0x0f000200,0x02008248,0x07c00e48,
+0x0c7fe780,0x04c177df,0x07802889,0x0780270a,0x0400228a,0x0b407289,
+0x04402209,0x0440128a,0x07802788,0x0b403289,0x04401208,0x04401208,
+0x0680c009,0x02009209,0x07400262,0x0a21009f,0x06800021,0x0c7fb300,
+0x00000000,0x07802888,0x0680c009,0x02009209,0x0c7fb1c0,0x07000261,
+0x07802889,0x0780270a,0x0400228a,0x0b407289,0x04402209,0x0440128a,
+0x07802788,0x0b403289,0x04401208,0x04401208,0x0680c009,0x05801288,
+0x02009289,0x0700024a,0x0a20c048,0x0c7800c0,0x094082a2,0x095102a2,
+0x0740024a,0x0a21009f,0x06800021,0x0c7fac00,0x00000000,0x07802888,
+0x0680c009,0x05801288,0x02009289,0x0700024a,0x0a20c048,0x0911084a,
+0x0900884a,0x0c7fa980,0x00000000,0x07802889,0x0780270a,0x0400228a,
 0x0b407289,0x04402209,0x0440128a,0x07802788,0x0b403289,0x04401208,
-0x04401208,0x0680c009,0x02009209,0x07400262,0x0a21009f,0x06800021,
-0x0c7fb340,0x00000000,0x07802888,0x0680c009,0x02009209,0x0c7fb200,
-0x07000261,0x07802889,0x0780270a,0x0400228a,0x0b407289,0x04402209,
-0x0440128a,0x07802788,0x0b403289,0x04401208,0x04401208,0x0680c009,
-0x05801288,0x02009289,0x0700024a,0x0a20c048,0x0c7800c0,0x094082a2,
-0x095102a2,0x0740024a,0x0a21009f,0x06800021,0x0c7fac40,0x00000000,
-0x07802888,0x0680c009,0x05801288,0x02009289,0x0700024a,0x0a20c048,
-0x0911084a,0x0900884a,0x0c7fa9c0,0x00000000,0x07802889,0x0780270a,
-0x0400228a,0x0b407289,0x04402209,0x0440128a,0x07802788,0x0b403289,
-0x04401208,0x04401208,0x0680c009,0x05802288,0x02009289,0x0700024a,
-0x09002208,0x0a60c008,0x0c780240,0x095842a2,0x0a60c048,0x0c780180,
-0x095042a2,0x0a60c088,0x0c7800c0,0x094842a2,0x094042a2,0x0740024a,
-0x0a21009f,0x06800021,0x0c7fa240,0x00000000,0x07802888,0x0680c009,
-0x05802288,0x02009289,0x0700024a,0x09002208,0x0aa1c008,0x0918484a,
-0x0aa14048,0x0910484a,0x0aa0c088,0x0908484a,0x0900484a,0x0c7f9e80,
-0x00000000,0x0c780080,0x06800021,0x0c7f9d80,0x00000000,0x06412107,
-0x090c11c7,0x0a6f8007,0x0c7dde00,0x00000000,0x06804009,0x06012d09,
-0x06800f8a,0x08012e00,0x00000000,0x0d07ff8a,0x00000000,0x0cc00000,
-0x00000000,0x06804009,0x06012c09,0x06809009,0x06012d09,0x06800f8a,
-0x06412e09,0x0d07ffca,0x06012e09,0x0cc00000,0x00000000,0x07800e49,
-0x0a60c009,0x0cc00000,0x00000000,0x06809009,0x06012c09,0x06804009,
-0x06012d09,0x06800f8a,0x06412e09,0x0d07ffca,0x06012e09,0x06411109,
-0x07802408,0x09446248,0x06011109,0x0cc00000,0x00000000,0x07800dc8,
-0x07801a09,0x0a608009,0x0c780880,0x0aa18049,0x0aa20089,0x0aa640c9,
-0x0a608109,0x0c7806c0,0x0c780000,0x05801288,0x0c780680,0x07c01a8a,
+0x04401208,0x0680c009,0x05802288,0x02009289,0x0700024a,0x09002208,
+0x0a60c008,0x0c780240,0x095842a2,0x0a60c048,0x0c780180,0x095042a2,
+0x0a60c088,0x0c7800c0,0x094842a2,0x094042a2,0x0740024a,0x0a21009f,
+0x06800021,0x0c7fa200,0x00000000,0x07802888,0x0680c009,0x05802288,
+0x02009289,0x0700024a,0x09002208,0x0aa1c008,0x0918484a,0x0aa14048,
+0x0910484a,0x0aa0c088,0x0908484a,0x0900484a,0x0c7f9e40,0x00000000,
+0x0c780080,0x06800021,0x0c7f9d40,0x00000000,0x06412107,0x090c11c7,
+0x0a6f8007,0x0c7de140,0x00000000,0x06804009,0x06012d09,0x06800f8a,
+0x08012e00,0x00000000,0x0d07ff8a,0x00000000,0x0cc00000,0x00000000,
+0x06804009,0x06012c09,0x06809009,0x06012d09,0x06800f8a,0x06412e09,
+0x0d07ffca,0x06012e09,0x0cc00000,0x00000000,0x07800e49,0x0a60c009,
+0x0cc00000,0x00000000,0x06809009,0x06012c09,0x06804009,0x06012d09,
+0x06800f8a,0x06412e09,0x0d07ffca,0x06012e09,0x06411109,0x07802408,
+0x09446248,0x06011109,0x0cc00000,0x00000000,0x07800dc8,0x07801a09,
+0x0a608009,0x0c780880,0x0aa18049,0x0aa20089,0x0aa640c9,0x0a608109,
+0x0c7806c0,0x0c780000,0x05801288,0x0c780680,0x07c01a8a,0x05802288,
+0x0540124a,0x02009289,0x0b0fe209,0x0400128a,0x0440224a,0x07c01a89,
+0x05401289,0x0400128a,0x0540124a,0x02009289,0x05401208,0x0b003209,
+0x07801a89,0x0440128a,0x0240a24a,0x0c780200,0x07c01aca,0x0c000b40,
+0x06801a8c,0x0c780100,0x00000000,0x0c000dc0,0x06801a8c,0x07800e08,
+0x07801a49,0x0a608009,0x0c780880,0x0aa18049,0x0aa20089,0x0aa640c9,
+0x0a608109,0x0c7806c0,0x0c780000,0x05801288,0x0c780680,0x07c01b8a,
 0x05802288,0x0540124a,0x02009289,0x0b0fe209,0x0400128a,0x0440224a,
-0x07c01a89,0x05401289,0x0400128a,0x0540124a,0x02009289,0x05401208,
-0x0b003209,0x07801a89,0x0440128a,0x0240a24a,0x0c780200,0x07c01aca,
-0x0c000b40,0x06801a8c,0x0c780100,0x00000000,0x0c000dc0,0x06801a8c,
-0x07800e08,0x07801a49,0x0a608009,0x0c780880,0x0aa18049,0x0aa20089,
-0x0aa640c9,0x0a608109,0x0c7806c0,0x0c780000,0x05801288,0x0c780680,
-0x07c01b8a,0x05802288,0x0540124a,0x02009289,0x0b0fe209,0x0400128a,
-0x0440224a,0x07c01b89,0x05401289,0x0400128a,0x0540124a,0x02009289,
-0x05401208,0x0b003209,0x07801b89,0x0440128a,0x0240a24a,0x0c780200,
-0x07c01bca,0x0c000200,0x06801b8c,0x0cc00000,0x00000000,0x0c000480,
-0x06801b8c,0x0cc00000,0x00000000,0x05802348,0x0740030d,0x05801248,
-0x0240a349,0x0400130c,0x0740030a,0x054012c8,0x0200b20b,0x058022cb,
-0x0240a24b,0x0400130c,0x0740030a,0x0cc00000,0x00000000,0x05803288,
-0x0540224a,0x02009289,0x0b0fe209,0x0400128a,0x0440224a,0x07400309,
-0x054012c8,0x04000349,0x0540224a,0x02009289,0x0b0fe2c9,0x0400128a,
-0x0440224a,0x0240a349,0x0400130c,0x0740030a,0x054012c8,0x0200b20b,
+0x07c01b89,0x05401289,0x0400128a,0x0540124a,0x02009289,0x05401208,
+0x0b003209,0x07801b89,0x0440128a,0x0240a24a,0x0c780200,0x07c01bca,
+0x0c000200,0x06801b8c,0x0cc00000,0x00000000,0x0c000480,0x06801b8c,
+0x0cc00000,0x00000000,0x05802348,0x0740030d,0x05801248,0x0240a349,
+0x0400130c,0x0740030a,0x054012c8,0x0200b20b,0x058022cb,0x0240a24b,
+0x0400130c,0x0740030a,0x0cc00000,0x00000000,0x05803288,0x0540224a,
+0x02009289,0x0b0fe209,0x0400128a,0x0440224a,0x07400309,0x054012c8,
 0x04000349,0x0540224a,0x02009289,0x0b0fe2c9,0x0400128a,0x0440224a,
-0x0240a349,0x0400130c,0x0740030a,0x054022c8,0x04000349,0x0540224a,
-0x02009289,0x0b0fe2c9,0x0400128a,0x0440224a,0x0240a349,0x0400130c,
-0x0740030a,0x0cc00000,0x00000000,0x0640c308,0x02008308,0x06035108,
-0x06800808,0x06035208,0x06800149,0x0680c00a,0x0700024b,0x0740028b,
-0x0b6105c9,0x04001249,0x0c7fff00,0x0400128a,0x06a0c008,0x06035008,
+0x0240a349,0x0400130c,0x0740030a,0x054012c8,0x0200b20b,0x04000349,
+0x0540224a,0x02009289,0x0b0fe2c9,0x0400128a,0x0440224a,0x0240a349,
+0x0400130c,0x0740030a,0x054022c8,0x04000349,0x0540224a,0x02009289,
+0x0b0fe2c9,0x0400128a,0x0440224a,0x0240a349,0x0400130c,0x0740030a,
+0x0cc00000,0x00000000,0x0640c308,0x02008308,0x06035108,0x06800808,
+0x06035208,0x06800149,0x0680c00a,0x0700024b,0x0740028b,0x0b6105c9,
+0x04001249,0x0c7fff00,0x0400128a,0x06a0c008,0x06035008,0x06435008,
+0x0580f208,0x0bef8048,0x00000000,0x06435308,0x0900c208,0x0a6f8008,
+0x00000000,0x0cc00000,0x00000000,0x0640c408,0x02008308,0x06035108,
+0x06801008,0x06035208,0x06800609,0x0680c00a,0x0700024b,0x0740028b,
+0x0b611449,0x04001249,0x0c7fff00,0x0400128a,0x06a0c008,0x06035008,
 0x06435008,0x0580f208,0x0bef8048,0x00000000,0x06435308,0x0900c208,
-0x0a6f8008,0x00000000,0x0cc00000,0x00000000,0x0640c408,0x02008308,
-0x06035108,0x06801008,0x06035208,0x06800609,0x0680c00a,0x0700024b,
-0x0740028b,0x0b611449,0x04001249,0x0c7fff00,0x0400128a,0x06a0c008,
+0x0a6f8008,0x00000000,0x0cc00000,0x00000000,0x0640c308,0x02008308,
+0x06035108,0x06800808,0x06035208,0x06b0c008,0x06035008,0x06435008,
+0x0580f208,0x0bef8048,0x00000000,0x06800149,0x0680c00a,0x0700028b,
+0x0740024b,0x0b6105c9,0x04001249,0x0c7fff00,0x0400128a,0x0cc00000,
+0x00000000,0x0640c408,0x02008308,0x06035108,0x06801008,0x06035208,
+0x06b0c008,0x06035008,0x06435008,0x0580f208,0x0bef8048,0x00000000,
+0x06800609,0x0680c00a,0x0700028b,0x0740024b,0x0b611449,0x04001249,
+0x0c7fff00,0x0400128a,0x0cc00000,0x00000000,0x0640c508,0x02008308,
+0x06035108,0x06800c08,0x06035208,0x06801489,0x0680c00a,0x0700024b,
+0x0740028b,0x0b611ec9,0x04001249,0x0c7fff00,0x0400128a,0x06a0c008,
 0x06035008,0x06435008,0x0580f208,0x0bef8048,0x00000000,0x06435308,
-0x0900c208,0x0a6f8008,0x00000000,0x0cc00000,0x00000000,0x0640c308,
-0x02008308,0x06035108,0x06800808,0x06035208,0x06b0c008,0x06035008,
-0x06435008,0x0580f208,0x0bef8048,0x00000000,0x06800149,0x0680c00a,
-0x0700028b,0x0740024b,0x0b6105c9,0x04001249,0x0c7fff00,0x0400128a,
-0x0cc00000,0x00000000,0x0640c408,0x02008308,0x06035108,0x06801008,
-0x06035208,0x06b0c008,0x06035008,0x06435008,0x0580f208,0x0bef8048,
-0x00000000,0x06800609,0x0680c00a,0x0700028b,0x0740024b,0x0b611449,
-0x04001249,0x0c7fff00,0x0400128a,0x0cc00000,0x00000000,0x0640c508,
-0x02008308,0x06035108,0x06800c08,0x06035208,0x06801489,0x0680c00a,
-0x0700024b,0x0740028b,0x0b611ec9,0x04001249,0x0c7fff00,0x0400128a,
-0x06a0c008,0x06035008,0x06435008,0x0580f208,0x0bef8048,0x00000000,
-0x06435308,0x0900c208,0x0a6f8008,0x00000000,0x07800f08,0x0aa20008,
-0x0540330c,0x078014c8,0x0aa0c008,0x06800008,0x06900008,0x0c000100,
-0x0200c20c,0x0cc00000,0x00000000,0x0680000d,0x08070300,0x0680c00b,
-0x0680000a,0x06470408,0x06470409,0x09508209,0x074002c8,0x040012cb,
-0x0baeffca,0x0400128a,0x0aa3800d,0x00000000,0x06470508,0x07c0fe88,
-0x09210208,0x07c0fec8,0x06470608,0x07c0ff08,0x09210208,0x07c0ff48,
-0x06470708,0x07c0ff88,0x09210208,0x07c0ffc8,0x0640cd08,0x02008308,
-0x0540924d,0x02008248,0x06035108,0x06804008,0x06035208,0x06a0c008,
-0x06035008,0x06435008,0x0580f208,0x0bef8048,0x00000000,0x0ba0c04d,
-0x0cc00000,0x00000000,0x0c7ff640,0x0400134d,0x0640c508,0x02008308,
-0x06035108,0x06800c08,0x06035208,0x06b0c008,0x06035008,0x06435008,
-0x0580f208,0x0bef8048,0x00000000,0x06801489,0x0680c00a,0x0700028b,
-0x0740024b,0x0b611ec9,0x04001249,0x0c7fff00,0x0400128a,0x0c000e00,
-0x00000000,0x07800f08,0x0aa20008,0x0540330c,0x078014c8,0x0aa0c008,
-0x06800008,0x06900008,0x0c000100,0x0200c20c,0x0cc00000,0x00000000,
-0x0680000d,0x08070200,0x0640cd08,0x02008308,0x0540924d,0x02008248,
-0x06035108,0x06804008,0x06035208,0x06b0c008,0x06035008,0x06435008,
-0x0580f208,0x0bef8048,0x00000000,0x0680c00b,0x0680000a,0x070002c8,
-0x040012cb,0x06070408,0x09108248,0x06070409,0x0baeffca,0x0400128a,
-0x0aa3800d,0x00000000,0x0780fe88,0x0780fec9,0x09610209,0x06070508,
-0x0780ff08,0x0780ff49,0x09610209,0x06070608,0x0780ff88,0x0780ffc9,
-0x09610209,0x06070708,0x0ba0c04d,0x0cc00000,0x00000000,0x0c7ff640,
-0x0400134d,0x078015c8,0x07801749,0x09421209,0x07801709,0x09441209,
-0x07801949,0x09461209,0x07800ec9,0x09483209,0x07800e89,0x094e3209,
-0x07800f49,0x09741209,0x07801789,0x09761209,0x07800d09,0x078017ca,
-0x02409289,0x09782209,0x06011708,0x06412508,0x07800fc9,0x09401209,
-0x07801089,0x09482209,0x078010c9,0x094c2209,0x07801009,0x09504209,
-0x07801049,0x09584209,0x06012508,0x06470108,0x07800f09,0x09401209,
-0x09421209,0x09441209,0x07800d09,0x09482209,0x06070108,0x06440808,
-0x07801289,0x09401209,0x07801709,0x09421209,0x06040808,0x0cc00000,
+0x0900c208,0x0a6f8008,0x00000000,0x07800f08,0x0aa20008,0x0540330c,
+0x078014c8,0x0aa0c008,0x06800008,0x06900008,0x0c000100,0x0200c20c,
+0x0cc00000,0x00000000,0x0680000d,0x08070300,0x0680c00b,0x0680000a,
+0x06470408,0x06470409,0x09508209,0x074002c8,0x040012cb,0x0baeffca,
+0x0400128a,0x0aa3800d,0x00000000,0x06470508,0x07c0fe88,0x09210208,
+0x07c0fec8,0x06470608,0x07c0ff08,0x09210208,0x07c0ff48,0x06470708,
+0x07c0ff88,0x09210208,0x07c0ffc8,0x0640cd08,0x02008308,0x0540924d,
+0x02008248,0x06035108,0x06804008,0x06035208,0x06a0c008,0x06035008,
+0x06435008,0x0580f208,0x0bef8048,0x00000000,0x0ba0c04d,0x0cc00000,
+0x00000000,0x0c7ff640,0x0400134d,0x0640c508,0x02008308,0x06035108,
+0x06800c08,0x06035208,0x06b0c008,0x06035008,0x06435008,0x0580f208,
+0x0bef8048,0x00000000,0x06801489,0x0680c00a,0x0700028b,0x0740024b,
+0x0b611ec9,0x04001249,0x0c7fff00,0x0400128a,0x0c000e00,0x00000000,
+0x07800f08,0x0aa20008,0x0540330c,0x078014c8,0x0aa0c008,0x06800008,
+0x06900008,0x0c000100,0x0200c20c,0x0cc00000,0x00000000,0x0680000d,
+0x08070200,0x0640cd08,0x02008308,0x0540924d,0x02008248,0x06035108,
+0x06804008,0x06035208,0x06b0c008,0x06035008,0x06435008,0x0580f208,
+0x0bef8048,0x00000000,0x0680c00b,0x0680000a,0x070002c8,0x040012cb,
+0x06070408,0x09108248,0x06070409,0x0baeffca,0x0400128a,0x0aa3800d,
+0x00000000,0x0780fe88,0x0780fec9,0x09610209,0x06070508,0x0780ff08,
+0x0780ff49,0x09610209,0x06070608,0x0780ff88,0x0780ffc9,0x09610209,
+0x06070708,0x0ba0c04d,0x0cc00000,0x00000000,0x0c7ff640,0x0400134d,
+0x078015c8,0x07801749,0x09421209,0x07801709,0x09441209,0x07801949,
+0x09461209,0x07800ec9,0x09483209,0x07800e89,0x094e3209,0x07800f49,
+0x09741209,0x07801789,0x09761209,0x07800d09,0x078017ca,0x02409289,
+0x09782209,0x06011708,0x06412508,0x07800fc9,0x09401209,0x07801089,
+0x09482209,0x078010c9,0x094c2209,0x07801009,0x09504209,0x07801049,
+0x09584209,0x06800cc9,0x09706209,0x07800889,0x09442209,0x0aa18009,
+0x0680018a,0x04401249,0x0340a24a,0x0960420a,0x09808048,0x078008c9,
+0x097c2209,0x0aa18009,0x0680018a,0x04401249,0x0340a24a,0x0968420a,
+0x09808048,0x06012508,0x06470108,0x07800f09,0x09401209,0x09421209,
+0x09441209,0x07800d09,0x09482209,0x06070108,0x06470b08,0x07800889,
+0x09402209,0x078008c9,0x09482209,0x09502209,0x06070b08,0x06440808,
+0x07801289,0x09401209,0x07801709,0x09421209,0x06040808,0x06440c08,
+0x07800889,0x09402209,0x078008c9,0x09442209,0x06040c08,0x0cc00000,
 0x00000000,0x0c002dc0,0x00000000,0x07802208,0x07c04408,0x07802248,
 0x07c04448,0x07802048,0x07c04488,0x07802148,0x07c044c8,0x07802008,
 0x07c04508,0x07801fc8,0x07c04548,0x07800808,0x07c045c8,0x07800848,
@@ -516,16 +519,18 @@ const u32 vh265_mc[] __attribute__((aligned(8))) = {
 0x0740024e,0x0cc00000,0x00000000,0x0640c023,0x0a6fffe3,0x00000000,
 0x06410b23,0x098230a3,0x06010b23,0x06412123,0x0a2fc063,0x00000000,
 0x06410c23,0x092039a3,0x07c04da6,0x093269a3,0x07c04de6,0x06410d23,
-0x06410b23,0x06800067,0x0c000b00,0x04000a26,0x0a6308a6,0x00000000,
-0x06411423,0x090188e3,0x0640ce24,0x0a610023,0x0be0c064,0x05801924,
-0x0a610024,0x06440123,0x0a2fc063,0x00000000,0x08012101,0x06803c25,
-0x07803363,0x0aa2c826,0x0a24c063,0x0aa24866,0x0a2440a3,0x0aa1c8a6,
-0x0a23c123,0x0be0c223,0x0aa08526,0x0a6304e6,0x098230e3,0x07c03363,
-0x0aa24826,0x06800065,0x0aa1c866,0x068000a5,0x0aa148a6,0x068000e5,
-0x0b20c566,0x06800125,0x06803c25,0x068000a7,0x0c000180,0x04000a25,
-0x0600c025,0x08030504,0x0c7ff140,0x00000000,0x0600d128,0x078033a8,
-0x0aa18028,0x00000000,0x0600d027,0x08007401,0x0640d027,0x0a6fc027,
-0x00000000,0x0cc00000,0x00000000,0x00000000,0x00000000,0x00000000,
+0x06410b23,0x0600d326,0x06800067,0x0c000d40,0x04000a26,0x0a6308a6,
+0x00000000,0x06411423,0x090188e3,0x0640ce24,0x0a610023,0x0be0c064,
+0x05801924,0x0a610024,0x06440123,0x0a2fc063,0x00000000,0x08012101,
+0x0640d224,0x0a214064,0x00000000,0x0800c009,0x0c7ff6c0,0x08007401,
+0x0640d224,0x0be3c0a4,0x06803c25,0x078033a3,0x0aa30826,0x0a250063,
+0x0be28124,0x0aa24866,0x0a2440a3,0x0aa1c8a6,0x0a23c123,0x0be0c223,
+0x0aa08526,0x0a6304e6,0x098230e3,0x07c033a3,0x0aa24826,0x06800065,
+0x0aa1c866,0x068000a5,0x0aa148a6,0x068000e5,0x0b20c566,0x06800125,
+0x06803c25,0x068000a7,0x0c000180,0x04000a25,0x0600c025,0x08030504,
+0x0c7feec0,0x00000000,0x0600d128,0x078033e8,0x0aa18028,0x00000000,
+0x0600d027,0x08007401,0x0640d027,0x0a6fc027,0x00000000,0x0cc00000,
+0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,
 0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,
 0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,
 0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,0x00000000,
diff --git a/drivers/amlogic/amports/vh265.c b/drivers/amlogic/amports/vh265.c
index bdc3a252a3cb..4b8958dcaa65 100755
--- a/drivers/amlogic/amports/vh265.c
+++ b/drivers/amlogic/amports/vh265.c
@@ -91,7 +91,8 @@ static u32 error_watchdog_count;
 #define H265_DEBUG_PARAM        4
 #define H265_DEBUG_REG          8
 #define H265_DEBUG_UCODE        0x10
-#define H265_DEBUG_DIS_ERROR_PROC       0x10000
+#define H265_DEBUG_USE_AUTO_SKIP        0x20
+#define H265_DEBUG_ENA_LOC_ERROR_PROC       0x10000
 #define H265_DEBUG_DIS_SYS_ERROR_PROC   0x20000
 
 #define DEBUG_PTS
@@ -237,6 +238,7 @@ static int m_temporalId;
 #define HEVC_CODED_SLICE_SEGMENT_DAT         5
 #define HEVC_DUMP_LMEM				7
 #define HEVC_SLICE_SEGMENT_DONE  		8
+#define HEVC_NAL_SEARCH_DONE			9
 
 #define HEVC_DISCARD_NAL         0xf0
 #define HEVC_ACTION_ERROR        0xfe
@@ -330,6 +332,14 @@ h265 buffer management
 
 #define DEBUG_REG1              HEVC_ASSIST_SCRATCH_G
 #define DEBUG_REG2              HEVC_ASSIST_SCRATCH_H
+/* 
+ucode parser/search control
+bit 0:  0, header auto parse; 1, header manual parse
+bit 1:  0, auto skip for noneseamless stream; 1, no skip
+bit 2: valid when bit1==0;  0, auto skip nal before first vps/sps/pps/idr; 1, auto skip nal before first vps
+*/
+#define NAL_SEARCH_CTL		      HEVC_ASSIST_SCRATCH_I
+#define CUR_NAL_UNIT_TYPE       HEVC_ASSIST_SCRATCH_J
 
 #define MAX_INT 0x7FFFFFFF
 
@@ -670,7 +680,10 @@ typedef struct hevc_state_{
     PIC_t* col_pic;
     int skip_flag;
     int decode_idx;
-    unsigned char skip_p_b_flag;
+    unsigned char have_vps;
+    unsigned char have_sps;
+    unsigned char have_pps;
+    unsigned char have_valid_start_slice;
     unsigned char wait_buf;
     unsigned char error_flag;
 }hevc_stru_t;
@@ -708,7 +721,10 @@ static void hevc_init_stru(hevc_stru_t* hevc, BuffInfo_t* buf_spec_i, buff_t* mc
     hevc->col_pic = NULL;
     hevc->wait_buf = 0;
     hevc->error_flag = 0;
-    hevc->skip_p_b_flag = 1;
+    hevc->have_vps = 0;
+    hevc->have_sps = 0;
+    hevc->have_pps = 0;
+    hevc->have_valid_start_slice = 0;
     
     for(i=0; i<MAX_REF_PIC_NUM; i++){
         m_PIC[i].index = -1;
@@ -2080,18 +2096,6 @@ static int hevc_slice_segment_header_process(hevc_stru_t* hevc, param_t* rpm_par
     int     lcu_y_num_div;
     int     Col_ref         ;
     if(hevc->wait_buf == 0){
-        if(hevc->skip_p_b_flag && 
-            (rpm_param->p.m_nalUnitType != NAL_UNIT_CODED_SLICE_IDR) &&
-            (rpm_param->p.m_nalUnitType != NAL_UNIT_CODED_SLICE_IDR_N_LP)&&
-            ( rpm_param->p.m_nalUnitType != NAL_UNIT_CODED_SLICE_CRA) &&
-            ( rpm_param->p.m_nalUnitType != NAL_UNIT_CODED_SLICE_BLA) && 
-            ( rpm_param->p.m_nalUnitType != NAL_UNIT_CODED_SLICE_BLANT) &&
-            (rpm_param->p.m_nalUnitType != NAL_UNIT_CODED_SLICE_BLA_N_LP )
-            ){
-            printk("%s: skip nal %d\n", __func__, rpm_param->p.m_nalUnitType);
-            return 4;
-        }
-        hevc->skip_p_b_flag = 0;    
         hevc->m_temporalId = rpm_param->p.m_temporalId;
         hevc->m_nalUnitType = rpm_param->p.m_nalUnitType;
         if(hevc->m_nalUnitType == NAL_UNIT_EOS){ 
@@ -2681,6 +2685,54 @@ static irqreturn_t vh265_isr(int irq, void *dev_id)
 
         printk("%s: error handle\n", __func__);
         hevc->error_flag = 0;
+    }
+    else if(dec_status == HEVC_NAL_SEARCH_DONE){
+        int naltype = READ_HREG(CUR_NAL_UNIT_TYPE);
+        int parse_type = HEVC_DISCARD_NAL;
+        if(naltype == NAL_UNIT_VPS){
+            parse_type = HEVC_NAL_UNIT_VPS;
+            hevc->have_vps = 1;
+        }
+        else if(hevc->have_vps){
+            if(naltype == NAL_UNIT_SPS){
+                parse_type = HEVC_NAL_UNIT_SPS;
+                hevc->have_sps = 1;
+            }
+            else if(naltype == NAL_UNIT_PPS){
+                parse_type = HEVC_NAL_UNIT_PPS;
+                hevc->have_pps = 1;
+            }
+            else if(hevc->have_sps && hevc->have_pps){
+                if(
+                    (naltype == NAL_UNIT_CODED_SLICE_IDR) ||
+                    (naltype == NAL_UNIT_CODED_SLICE_IDR_N_LP)||
+                    ( naltype == NAL_UNIT_CODED_SLICE_CRA) ||
+                    ( naltype == NAL_UNIT_CODED_SLICE_BLA) || 
+                    ( naltype == NAL_UNIT_CODED_SLICE_BLANT) ||
+                    (naltype == NAL_UNIT_CODED_SLICE_BLA_N_LP )
+                 ){
+                    parse_type = HEVC_NAL_UNIT_CODED_SLICE_SEGMENT;
+                    hevc->have_valid_start_slice = 1;
+                }
+                else if(hevc->have_valid_start_slice&&(naltype<= NAL_UNIT_CODED_SLICE_CRA)) {
+                    parse_type = HEVC_NAL_UNIT_CODED_SLICE_SEGMENT;
+                }
+            }
+        }
+        if(hevc->have_vps && hevc->have_sps && hevc->have_pps && hevc->have_valid_start_slice){
+            WRITE_VREG(NAL_SEARCH_CTL, 0x2); //auot parser NAL; do not check vps/sps/pps/idr
+        }
+        
+        if(debug&H265_DEBUG_PARAM){
+            printk("naltype = %d  parse_type %d\n %d %d %d %d \n", naltype, parse_type,
+                hevc->have_vps ,hevc->have_sps, hevc->have_pps ,hevc->have_valid_start_slice);
+        }
+
+        WRITE_VREG(HEVC_DEC_STATUS_REG, parse_type);
+        // Interrupt Amrisc to excute 
+        WRITE_VREG(HEVC_MCPU_INTR_REQ, AMRISC_MAIN_REQ);
+            
+    
     }
     else if(dec_status == HEVC_SLICE_SEGMENT_DONE){
         if(hevc->wait_buf == 0){
@@ -2744,7 +2796,7 @@ static void vh265_put_timer_func(unsigned long arg)
         ) {                        // no buffer to recycle
         error_watchdog_count++;
         if (error_watchdog_count == ERROR_LOCAL_RESET_COUNT) {    
-            if((debug&H265_DEBUG_DIS_ERROR_PROC)==0){
+            if((debug&H265_DEBUG_ENA_LOC_ERROR_PROC)!=0){
                 printk("H265 decoder error local reset.\n");
                 gHevc.error_flag = 1;
                 WRITE_VREG(HEVC_ASSIST_MBOX1_IRQ_REG, 0x1); 
@@ -2829,6 +2881,11 @@ static void vh265_prot_init(void)
     else{
         WRITE_VREG(DEBUG_REG1, 0x0);
     }
+    
+    if(debug&H265_DEBUG_USE_AUTO_SKIP)
+        WRITE_VREG(NAL_SEARCH_CTL, 0x0); //auto parser NAL and skip
+    else    
+        WRITE_VREG(NAL_SEARCH_CTL, 0x1); //manual parser NAL
 
 }
 
-- 
2.19.0

